// proxycurl-js-linkedin-profile-scraper/dist/model/AcquiredCompany.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AcquiredCompanyAnnouncedDate = _interopRequireDefault(require("./AcquiredCompanyAnnouncedDate"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The AcquiredCompany model module.
 * @module model/AcquiredCompany
 * @version 1.0.0
 */
var AcquiredCompany = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AcquiredCompany</code>.
   * @alias module:model/AcquiredCompany
   */
  function AcquiredCompany() {
    _classCallCheck(this, AcquiredCompany);
    AcquiredCompany.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(AcquiredCompany, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>AcquiredCompany</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AcquiredCompany} obj Optional instance to populate.
     * @return {module:model/AcquiredCompany} The populated <code>AcquiredCompany</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AcquiredCompany();
        if (data.hasOwnProperty('linkedin_profile_url')) {
          obj['linkedin_profile_url'] = _ApiClient["default"].convertToType(data['linkedin_profile_url'], 'String');
        }
        if (data.hasOwnProperty('crunchbase_profile_url')) {
          obj['crunchbase_profile_url'] = _ApiClient["default"].convertToType(data['crunchbase_profile_url'], 'String');
        }
        if (data.hasOwnProperty('announced_date')) {
          obj['announced_date'] = _AcquiredCompanyAnnouncedDate["default"].constructFromObject(data['announced_date']);
        }
        if (data.hasOwnProperty('price')) {
          obj['price'] = _ApiClient["default"].convertToType(data['price'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>AcquiredCompany</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AcquiredCompany</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['linkedin_profile_url'] && !(typeof data['linkedin_profile_url'] === 'string' || data['linkedin_profile_url'] instanceof String)) {
        throw new Error("Expected the field `linkedin_profile_url` to be a primitive type in the JSON string but got " + data['linkedin_profile_url']);
      }
      // ensure the json data is a string
      if (data['crunchbase_profile_url'] && !(typeof data['crunchbase_profile_url'] === 'string' || data['crunchbase_profile_url'] instanceof String)) {
        throw new Error("Expected the field `crunchbase_profile_url` to be a primitive type in the JSON string but got " + data['crunchbase_profile_url']);
      }
      // validate the optional field `announced_date`
      if (data['announced_date']) {
        // data not null
        _AcquiredCompanyAnnouncedDate["default"].validateJSON(data['announced_date']);
      }
      return true;
    }
  }]);
  return AcquiredCompany;
}();
/**
 * LinkedIn Company Profile URL of company that was involved
 * @member {String} linkedin_profile_url
 */
AcquiredCompany.prototype['linkedin_profile_url'] = undefined;

/**
 * Crunchbase Profile URL of company that was involved
 * @member {String} crunchbase_profile_url
 */
AcquiredCompany.prototype['crunchbase_profile_url'] = undefined;

/**
 * @member {module:model/AcquiredCompanyAnnouncedDate} announced_date
 */
AcquiredCompany.prototype['announced_date'] = undefined;

/**
 * Price of acquisition
 * @member {Number} price
 */
AcquiredCompany.prototype['price'] = undefined;
var _default = AcquiredCompany;
exports["default"] = _default;
