// proxycurl-js-linkedin-profile-scraper/dist/model/Activity.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The Activity model module.
 * @module model/Activity
 * @version 1.0.0
 */
var Activity = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Activity</code>.
   * @alias module:model/Activity
   */
  function Activity() {
    _classCallCheck(this, Activity);
    Activity.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Activity, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Activity</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Activity} obj Optional instance to populate.
     * @return {module:model/Activity} The populated <code>Activity</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Activity();
        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
        }
        if (data.hasOwnProperty('link')) {
          obj['link'] = _ApiClient["default"].convertToType(data['link'], 'String');
        }
        if (data.hasOwnProperty('activity_status')) {
          obj['activity_status'] = _ApiClient["default"].convertToType(data['activity_status'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Activity</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Activity</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['title'] && !(typeof data['title'] === 'string' || data['title'] instanceof String)) {
        throw new Error("Expected the field `title` to be a primitive type in the JSON string but got " + data['title']);
      }
      // ensure the json data is a string
      if (data['link'] && !(typeof data['link'] === 'string' || data['link'] instanceof String)) {
        throw new Error("Expected the field `link` to be a primitive type in the JSON string but got " + data['link']);
      }
      // ensure the json data is a string
      if (data['activity_status'] && !(typeof data['activity_status'] === 'string' || data['activity_status'] instanceof String)) {
        throw new Error("Expected the field `activity_status` to be a primitive type in the JSON string but got " + data['activity_status']);
      }
      return true;
    }
  }]);
  return Activity;
}();
/**
 * @member {String} title
 */
Activity.prototype['title'] = undefined;

/**
 * @member {String} link
 */
Activity.prototype['link'] = undefined;

/**
 * @member {String} activity_status
 */
Activity.prototype['activity_status'] = undefined;
var _default = Activity;
exports["default"] = _default;
