// proxycurl-js-linkedin-profile-scraper/dist/model/Article.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _CompanyUpdatePostedOn = _interopRequireDefault(require("./CompanyUpdatePostedOn"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The Article model module.
 * @module model/Article
 * @version 1.0.0
 */
var Article = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Article</code>.
   * @alias module:model/Article
   */
  function Article() {
    _classCallCheck(this, Article);
    Article.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Article, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Article</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Article} obj Optional instance to populate.
     * @return {module:model/Article} The populated <code>Article</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Article();
        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
        }
        if (data.hasOwnProperty('link')) {
          obj['link'] = _ApiClient["default"].convertToType(data['link'], 'String');
        }
        if (data.hasOwnProperty('published_date')) {
          obj['published_date'] = _CompanyUpdatePostedOn["default"].constructFromObject(data['published_date']);
        }
        if (data.hasOwnProperty('author')) {
          obj['author'] = _ApiClient["default"].convertToType(data['author'], 'String');
        }
        if (data.hasOwnProperty('image_url')) {
          obj['image_url'] = _ApiClient["default"].convertToType(data['image_url'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Article</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Article</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['title'] && !(typeof data['title'] === 'string' || data['title'] instanceof String)) {
        throw new Error("Expected the field `title` to be a primitive type in the JSON string but got " + data['title']);
      }
      // ensure the json data is a string
      if (data['link'] && !(typeof data['link'] === 'string' || data['link'] instanceof String)) {
        throw new Error("Expected the field `link` to be a primitive type in the JSON string but got " + data['link']);
      }
      // validate the optional field `published_date`
      if (data['published_date']) {
        // data not null
        _CompanyUpdatePostedOn["default"].validateJSON(data['published_date']);
      }
      // ensure the json data is a string
      if (data['author'] && !(typeof data['author'] === 'string' || data['author'] instanceof String)) {
        throw new Error("Expected the field `author` to be a primitive type in the JSON string but got " + data['author']);
      }
      // ensure the json data is a string
      if (data['image_url'] && !(typeof data['image_url'] === 'string' || data['image_url'] instanceof String)) {
        throw new Error("Expected the field `image_url` to be a primitive type in the JSON string but got " + data['image_url']);
      }
      return true;
    }
  }]);
  return Article;
}();
/**
 * @member {String} title
 */
Article.prototype['title'] = undefined;

/**
 * @member {String} link
 */
Article.prototype['link'] = undefined;

/**
 * @member {module:model/CompanyUpdatePostedOn} published_date
 */
Article.prototype['published_date'] = undefined;

/**
 * @member {String} author
 */
Article.prototype['author'] = undefined;

/**
 * @member {String} image_url
 */
Article.prototype['image_url'] = undefined;
var _default = Article;
exports["default"] = _default;
