// proxycurl-js-linkedin-profile-scraper/dist/model/CompanyUpdate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _CompanyUpdatePostedOn = _interopRequireDefault(require("./CompanyUpdatePostedOn"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The CompanyUpdate model module.
 * @module model/CompanyUpdate
 * @version 1.0.0
 */
var CompanyUpdate = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CompanyUpdate</code>.
   * @alias module:model/CompanyUpdate
   */
  function CompanyUpdate() {
    _classCallCheck(this, CompanyUpdate);
    CompanyUpdate.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(CompanyUpdate, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>CompanyUpdate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CompanyUpdate} obj Optional instance to populate.
     * @return {module:model/CompanyUpdate} The populated <code>CompanyUpdate</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CompanyUpdate();
        if (data.hasOwnProperty('article_link')) {
          obj['article_link'] = _ApiClient["default"].convertToType(data['article_link'], 'String');
        }
        if (data.hasOwnProperty('image')) {
          obj['image'] = _ApiClient["default"].convertToType(data['image'], 'String');
        }
        if (data.hasOwnProperty('posted_on')) {
          obj['posted_on'] = _CompanyUpdatePostedOn["default"].constructFromObject(data['posted_on']);
        }
        if (data.hasOwnProperty('text')) {
          obj['text'] = _ApiClient["default"].convertToType(data['text'], 'String');
        }
        if (data.hasOwnProperty('total_likes')) {
          obj['total_likes'] = _ApiClient["default"].convertToType(data['total_likes'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>CompanyUpdate</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CompanyUpdate</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['article_link'] && !(typeof data['article_link'] === 'string' || data['article_link'] instanceof String)) {
        throw new Error("Expected the field `article_link` to be a primitive type in the JSON string but got " + data['article_link']);
      }
      // ensure the json data is a string
      if (data['image'] && !(typeof data['image'] === 'string' || data['image'] instanceof String)) {
        throw new Error("Expected the field `image` to be a primitive type in the JSON string but got " + data['image']);
      }
      // validate the optional field `posted_on`
      if (data['posted_on']) {
        // data not null
        _CompanyUpdatePostedOn["default"].validateJSON(data['posted_on']);
      }
      // ensure the json data is a string
      if (data['text'] && !(typeof data['text'] === 'string' || data['text'] instanceof String)) {
        throw new Error("Expected the field `text` to be a primitive type in the JSON string but got " + data['text']);
      }
      return true;
    }
  }]);
  return CompanyUpdate;
}();
/**
 * The URL for which the post links out to
 * @member {String} article_link
 */
CompanyUpdate.prototype['article_link'] = undefined;

/**
 * The URL to the image to the post (if it exists)
 * @member {String} image
 */
CompanyUpdate.prototype['image'] = undefined;

/**
 * @member {module:model/CompanyUpdatePostedOn} posted_on
 */
CompanyUpdate.prototype['posted_on'] = undefined;

/**
 * The body of the update
 * @member {String} text
 */
CompanyUpdate.prototype['text'] = undefined;

/**
 * The total likes a post has received
 * @member {Number} total_likes
 */
CompanyUpdate.prototype['total_likes'] = undefined;
var _default = CompanyUpdate;
exports["default"] = _default;
