// proxycurl-js-linkedin-profile-scraper/dist/model/CompanyReveal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _CompanyRevealCompany = _interopRequireDefault(require("./CompanyRevealCompany"));
var _CompanyRevealRoleProfile = _interopRequireDefault(require("./CompanyRevealRoleProfile"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The CompanyReveal model module.
 * @module model/CompanyReveal
 * @version 1.0.0
 */
var CompanyReveal = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CompanyReveal</code>.
   * @alias module:model/CompanyReveal
   */
  function CompanyReveal() {
    _classCallCheck(this, CompanyReveal);
    CompanyReveal.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(CompanyReveal, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>CompanyReveal</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CompanyReveal} obj Optional instance to populate.
     * @return {module:model/CompanyReveal} The populated <code>CompanyReveal</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CompanyReveal();
        if (data.hasOwnProperty('company')) {
          obj['company'] = _CompanyRevealCompany["default"].constructFromObject(data['company']);
        }
        if (data.hasOwnProperty('company_linkedin_profile_url')) {
          obj['company_linkedin_profile_url'] = _ApiClient["default"].convertToType(data['company_linkedin_profile_url'], 'String');
        }
        if (data.hasOwnProperty('role_contact_number')) {
          obj['role_contact_number'] = _ApiClient["default"].convertToType(data['role_contact_number'], ['String']);
        }
        if (data.hasOwnProperty('role_personal_email')) {
          obj['role_personal_email'] = _ApiClient["default"].convertToType(data['role_personal_email'], ['String']);
        }
        if (data.hasOwnProperty('role_profile')) {
          obj['role_profile'] = _CompanyRevealRoleProfile["default"].constructFromObject(data['role_profile']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>CompanyReveal</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CompanyReveal</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `company`
      if (data['company']) {
        // data not null
        _CompanyRevealCompany["default"].validateJSON(data['company']);
      }
      // ensure the json data is a string
      if (data['company_linkedin_profile_url'] && !(typeof data['company_linkedin_profile_url'] === 'string' || data['company_linkedin_profile_url'] instanceof String)) {
        throw new Error("Expected the field `company_linkedin_profile_url` to be a primitive type in the JSON string but got " + data['company_linkedin_profile_url']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['role_contact_number'])) {
        throw new Error("Expected the field `role_contact_number` to be an array in the JSON data but got " + data['role_contact_number']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['role_personal_email'])) {
        throw new Error("Expected the field `role_personal_email` to be an array in the JSON data but got " + data['role_personal_email']);
      }
      // validate the optional field `role_profile`
      if (data['role_profile']) {
        // data not null
        _CompanyRevealRoleProfile["default"].validateJSON(data['role_profile']);
      }
      return true;
    }
  }]);
  return CompanyReveal;
}();
/**
 * @member {module:model/CompanyRevealCompany} company
 */
CompanyReveal.prototype['company'] = undefined;

/**
 * LinkedIn Profile URL of the Company returned.
 * @member {String} company_linkedin_profile_url
 */
CompanyReveal.prototype['company_linkedin_profile_url'] = undefined;

/**
 * A list of personal contact numbers.
 * @member {Array.<String>} role_contact_number
 */
CompanyReveal.prototype['role_contact_number'] = undefined;

/**
 * A list of personal email addresses.
 * @member {Array.<String>} role_personal_email
 */
CompanyReveal.prototype['role_personal_email'] = undefined;

/**
 * @member {module:model/CompanyRevealRoleProfile} role_profile
 */
CompanyReveal.prototype['role_profile'] = undefined;
var _default = CompanyReveal;
exports["default"] = _default;
