// proxycurl-js-linkedin-profile-scraper/dist/model/Course.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The Course model module.
 * @module model/Course
 * @version 1.0.0
 */
var Course = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Course</code>.
   * @alias module:model/Course
   */
  function Course() {
    _classCallCheck(this, Course);
    Course.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Course, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Course</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Course} obj Optional instance to populate.
     * @return {module:model/Course} The populated <code>Course</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Course();
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('number')) {
          obj['number'] = _ApiClient["default"].convertToType(data['number'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Course</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Course</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['number'] && !(typeof data['number'] === 'string' || data['number'] instanceof String)) {
        throw new Error("Expected the field `number` to be a primitive type in the JSON string but got " + data['number']);
      }
      return true;
    }
  }]);
  return Course;
}();
/**
 * Name of the course
 * @member {String} name
 */
Course.prototype['name'] = undefined;

/**
 * The numerical representation of the course
 * @member {String} number
 */
Course.prototype['number'] = undefined;
var _default = Course;
exports["default"] = _default;
