// proxycurl-js-linkedin-profile-scraper/dist/model/Employee.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _EmployeeProfile = _interopRequireDefault(require("./EmployeeProfile"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The Employee model module.
 * @module model/Employee
 * @version 1.0.0
 */
var Employee = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Employee</code>.
   * @alias module:model/Employee
   */
  function Employee() {
    _classCallCheck(this, Employee);
    Employee.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Employee, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Employee</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Employee} obj Optional instance to populate.
     * @return {module:model/Employee} The populated <code>Employee</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Employee();
        if (data.hasOwnProperty('profile_url')) {
          obj['profile_url'] = _ApiClient["default"].convertToType(data['profile_url'], 'String');
        }
        if (data.hasOwnProperty('profile')) {
          obj['profile'] = _EmployeeProfile["default"].constructFromObject(data['profile']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Employee</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Employee</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['profile_url'] && !(typeof data['profile_url'] === 'string' || data['profile_url'] instanceof String)) {
        throw new Error("Expected the field `profile_url` to be a primitive type in the JSON string but got " + data['profile_url']);
      }
      // validate the optional field `profile`
      if (data['profile']) {
        // data not null
        _EmployeeProfile["default"].validateJSON(data['profile']);
      }
      return true;
    }
  }]);
  return Employee;
}();
/**
 * @member {String} profile_url
 */
Employee.prototype['profile_url'] = undefined;

/**
 * @member {module:model/EmployeeProfile} profile
 */
Employee.prototype['profile'] = undefined;
var _default = Employee;
exports["default"] = _default;
