// proxycurl-js-linkedin-profile-scraper/dist/model/Funding.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _FundingAnnouncedDate = _interopRequireDefault(require("./FundingAnnouncedDate"));
var _Investor = _interopRequireDefault(require("./Investor"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The Funding model module.
 * @module model/Funding
 * @version 1.0.0
 */
var Funding = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Funding</code>.
   * @alias module:model/Funding
   */
  function Funding() {
    _classCallCheck(this, Funding);
    Funding.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Funding, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Funding</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Funding} obj Optional instance to populate.
     * @return {module:model/Funding} The populated <code>Funding</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Funding();
        if (data.hasOwnProperty('funding_type')) {
          obj['funding_type'] = _ApiClient["default"].convertToType(data['funding_type'], 'String');
        }
        if (data.hasOwnProperty('money_raised')) {
          obj['money_raised'] = _ApiClient["default"].convertToType(data['money_raised'], 'Number');
        }
        if (data.hasOwnProperty('announced_date')) {
          obj['announced_date'] = _FundingAnnouncedDate["default"].constructFromObject(data['announced_date']);
        }
        if (data.hasOwnProperty('number_of_investor')) {
          obj['number_of_investor'] = _ApiClient["default"].convertToType(data['number_of_investor'], 'Number');
        }
        if (data.hasOwnProperty('investor_list')) {
          obj['investor_list'] = _ApiClient["default"].convertToType(data['investor_list'], [_Investor["default"]]);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Funding</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Funding</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['funding_type'] && !(typeof data['funding_type'] === 'string' || data['funding_type'] instanceof String)) {
        throw new Error("Expected the field `funding_type` to be a primitive type in the JSON string but got " + data['funding_type']);
      }
      // validate the optional field `announced_date`
      if (data['announced_date']) {
        // data not null
        _FundingAnnouncedDate["default"].validateJSON(data['announced_date']);
      }
      if (data['investor_list']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['investor_list'])) {
          throw new Error("Expected the field `investor_list` to be an array in the JSON data but got " + data['investor_list']);
        }
        // validate the optional field `investor_list` (array)
        var _iterator = _createForOfIteratorHelper(data['investor_list']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _Investor["default"].validateJsonObject(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      return true;
    }
  }]);
  return Funding;
}();
/**
 * Type of funding
 * @member {String} funding_type
 */
Funding.prototype['funding_type'] = undefined;

/**
 * Amount of money raised
 * @member {Number} money_raised
 */
Funding.prototype['money_raised'] = undefined;

/**
 * @member {module:model/FundingAnnouncedDate} announced_date
 */
Funding.prototype['announced_date'] = undefined;

/**
 * Number of investors in this round
 * @member {Number} number_of_investor
 */
Funding.prototype['number_of_investor'] = undefined;

/**
 * @member {Array.<module:model/Investor>} investor_list
 */
Funding.prototype['investor_list'] = undefined;
var _default = Funding;
exports["default"] = _default;
