// proxycurl-js-linkedin-profile-scraper/dist/model/HonourAward.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _HonourAwardIssuedOn = _interopRequireDefault(require("./HonourAwardIssuedOn"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The HonourAward model module.
 * @module model/HonourAward
 * @version 1.0.0
 */
var HonourAward = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>HonourAward</code>.
   * @alias module:model/HonourAward
   */
  function HonourAward() {
    _classCallCheck(this, HonourAward);
    HonourAward.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(HonourAward, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>HonourAward</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/HonourAward} obj Optional instance to populate.
     * @return {module:model/HonourAward} The populated <code>HonourAward</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new HonourAward();
        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
        }
        if (data.hasOwnProperty('issuer')) {
          obj['issuer'] = _ApiClient["default"].convertToType(data['issuer'], 'String');
        }
        if (data.hasOwnProperty('issued_on')) {
          obj['issued_on'] = _HonourAwardIssuedOn["default"].constructFromObject(data['issued_on']);
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>HonourAward</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>HonourAward</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['title'] && !(typeof data['title'] === 'string' || data['title'] instanceof String)) {
        throw new Error("Expected the field `title` to be a primitive type in the JSON string but got " + data['title']);
      }
      // ensure the json data is a string
      if (data['issuer'] && !(typeof data['issuer'] === 'string' || data['issuer'] instanceof String)) {
        throw new Error("Expected the field `issuer` to be a primitive type in the JSON string but got " + data['issuer']);
      }
      // validate the optional field `issued_on`
      if (data['issued_on']) {
        // data not null
        _HonourAwardIssuedOn["default"].validateJSON(data['issued_on']);
      }
      // ensure the json data is a string
      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      }
      return true;
    }
  }]);
  return HonourAward;
}();
/**
 * Title of the honour/award.
 * @member {String} title
 */
HonourAward.prototype['title'] = undefined;

/**
 * The organisation body issuing this honour/award.
 * @member {String} issuer
 */
HonourAward.prototype['issuer'] = undefined;

/**
 * @member {module:model/HonourAwardIssuedOn} issued_on
 */
HonourAward.prototype['issued_on'] = undefined;

/**
 * Description of the honour/award.
 * @member {String} description
 */
HonourAward.prototype['description'] = undefined;
var _default = HonourAward;
exports["default"] = _default;
