// proxycurl-js-linkedin-profile-scraper/dist/model/InferredSalary.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The InferredSalary model module.
 * @module model/InferredSalary
 * @version 1.0.0
 */
var InferredSalary = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>InferredSalary</code>.
   * @alias module:model/InferredSalary
   */
  function InferredSalary() {
    _classCallCheck(this, InferredSalary);
    InferredSalary.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(InferredSalary, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>InferredSalary</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/InferredSalary} obj Optional instance to populate.
     * @return {module:model/InferredSalary} The populated <code>InferredSalary</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new InferredSalary();
        if (data.hasOwnProperty('min')) {
          obj['min'] = _ApiClient["default"].convertToType(data['min'], 'Number');
        }
        if (data.hasOwnProperty('max')) {
          obj['max'] = _ApiClient["default"].convertToType(data['max'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>InferredSalary</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>InferredSalary</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      return true;
    }
  }]);
  return InferredSalary;
}();
/**
 * @member {Number} min
 */
InferredSalary.prototype['min'] = undefined;

/**
 * @member {Number} max
 */
InferredSalary.prototype['max'] = undefined;
var _default = InferredSalary;
exports["default"] = _default;
