// proxycurl-js-linkedin-profile-scraper/dist/model/Investor.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The Investor model module.
 * @module model/Investor
 * @version 1.0.0
 */
var Investor = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Investor</code>.
   * @alias module:model/Investor
   */
  function Investor() {
    _classCallCheck(this, Investor);
    Investor.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Investor, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Investor</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Investor} obj Optional instance to populate.
     * @return {module:model/Investor} The populated <code>Investor</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Investor();
        if (data.hasOwnProperty('linkedin_profile_url')) {
          obj['linkedin_profile_url'] = _ApiClient["default"].convertToType(data['linkedin_profile_url'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Investor</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Investor</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['linkedin_profile_url'] && !(typeof data['linkedin_profile_url'] === 'string' || data['linkedin_profile_url'] instanceof String)) {
        throw new Error("Expected the field `linkedin_profile_url` to be a primitive type in the JSON string but got " + data['linkedin_profile_url']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['type'] && !(typeof data['type'] === 'string' || data['type'] instanceof String)) {
        throw new Error("Expected the field `type` to be a primitive type in the JSON string but got " + data['type']);
      }
      return true;
    }
  }]);
  return Investor;
}();
/**
 * LinkedIn Profile URL of investor
 * @member {String} linkedin_profile_url
 */
Investor.prototype['linkedin_profile_url'] = undefined;

/**
 * Name of investor
 * @member {String} name
 */
Investor.prototype['name'] = undefined;

/**
 * Type of investor
 * @member {String} type
 */
Investor.prototype['type'] = undefined;
var _default = Investor;
exports["default"] = _default;
