// proxycurl-js-linkedin-profile-scraper/dist/model/JobCompany.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The JobCompany model module.
 * @module model/JobCompany
 * @version 1.0.0
 */
var JobCompany = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>JobCompany</code>.
   * @alias module:model/JobCompany
   */
  function JobCompany() {
    _classCallCheck(this, JobCompany);
    JobCompany.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(JobCompany, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>JobCompany</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/JobCompany} obj Optional instance to populate.
     * @return {module:model/JobCompany} The populated <code>JobCompany</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new JobCompany();
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('url')) {
          obj['url'] = _ApiClient["default"].convertToType(data['url'], 'String');
        }
        if (data.hasOwnProperty('logo')) {
          obj['logo'] = _ApiClient["default"].convertToType(data['logo'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>JobCompany</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>JobCompany</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['url'] && !(typeof data['url'] === 'string' || data['url'] instanceof String)) {
        throw new Error("Expected the field `url` to be a primitive type in the JSON string but got " + data['url']);
      }
      // ensure the json data is a string
      if (data['logo'] && !(typeof data['logo'] === 'string' || data['logo'] instanceof String)) {
        throw new Error("Expected the field `logo` to be a primitive type in the JSON string but got " + data['logo']);
      }
      return true;
    }
  }]);
  return JobCompany;
}();
/**
 * @member {String} name
 */
JobCompany.prototype['name'] = undefined;

/**
 * @member {String} url
 */
JobCompany.prototype['url'] = undefined;

/**
 * @member {String} logo
 */
JobCompany.prototype['logo'] = undefined;
var _default = JobCompany;
exports["default"] = _default;
