// proxycurl-js-linkedin-profile-scraper/dist/model/JobListPage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Job = _interopRequireDefault(require("./Job"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The JobListPage model module.
 * @module model/JobListPage
 * @version 1.0.0
 */
var JobListPage = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>JobListPage</code>.
   * @alias module:model/JobListPage
   */
  function JobListPage() {
    _classCallCheck(this, JobListPage);
    JobListPage.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(JobListPage, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>JobListPage</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/JobListPage} obj Optional instance to populate.
     * @return {module:model/JobListPage} The populated <code>JobListPage</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new JobListPage();
        if (data.hasOwnProperty('job')) {
          obj['job'] = _ApiClient["default"].convertToType(data['job'], [_Job["default"]]);
        }
        if (data.hasOwnProperty('next_page_no')) {
          obj['next_page_no'] = _ApiClient["default"].convertToType(data['next_page_no'], 'Number');
        }
        if (data.hasOwnProperty('next_page_api_url')) {
          obj['next_page_api_url'] = _ApiClient["default"].convertToType(data['next_page_api_url'], 'String');
        }
        if (data.hasOwnProperty('previous_page_no')) {
          obj['previous_page_no'] = _ApiClient["default"].convertToType(data['previous_page_no'], 'Number');
        }
        if (data.hasOwnProperty('previous_page_api_url')) {
          obj['previous_page_api_url'] = _ApiClient["default"].convertToType(data['previous_page_api_url'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>JobListPage</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>JobListPage</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['job']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['job'])) {
          throw new Error("Expected the field `job` to be an array in the JSON data but got " + data['job']);
        }
        // validate the optional field `job` (array)
        var _iterator = _createForOfIteratorHelper(data['job']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _Job["default"].validateJsonObject(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['next_page_api_url'] && !(typeof data['next_page_api_url'] === 'string' || data['next_page_api_url'] instanceof String)) {
        throw new Error("Expected the field `next_page_api_url` to be a primitive type in the JSON string but got " + data['next_page_api_url']);
      }
      // ensure the json data is a string
      if (data['previous_page_api_url'] && !(typeof data['previous_page_api_url'] === 'string' || data['previous_page_api_url'] instanceof String)) {
        throw new Error("Expected the field `previous_page_api_url` to be a primitive type in the JSON string but got " + data['previous_page_api_url']);
      }
      return true;
    }
  }]);
  return JobListPage;
}();
/**
 * @member {Array.<module:model/Job>} job
 */
JobListPage.prototype['job'] = undefined;

/**
 * @member {Number} next_page_no
 */
JobListPage.prototype['next_page_no'] = undefined;

/**
 * @member {String} next_page_api_url
 */
JobListPage.prototype['next_page_api_url'] = undefined;

/**
 * @member {Number} previous_page_no
 */
JobListPage.prototype['previous_page_no'] = undefined;

/**
 * @member {String} previous_page_api_url
 */
JobListPage.prototype['previous_page_api_url'] = undefined;
var _default = JobListPage;
exports["default"] = _default;
