// proxycurl-js-linkedin-profile-scraper/dist/model/LinkedinCompanyAcquisitions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AcquiredCompany = _interopRequireDefault(require("./AcquiredCompany"));
var _Acquisition = _interopRequireDefault(require("./Acquisition"));
var _AcquisitionAcquiredBy = _interopRequireDefault(require("./AcquisitionAcquiredBy"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The LinkedinCompanyAcquisitions model module.
 * @module model/LinkedinCompanyAcquisitions
 * @version 1.0.0
 */
var LinkedinCompanyAcquisitions = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>LinkedinCompanyAcquisitions</code>.
   * @alias module:model/LinkedinCompanyAcquisitions
   * @implements module:model/Acquisition
   */
  function LinkedinCompanyAcquisitions() {
    _classCallCheck(this, LinkedinCompanyAcquisitions);
    _Acquisition["default"].initialize(this);
    LinkedinCompanyAcquisitions.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(LinkedinCompanyAcquisitions, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>LinkedinCompanyAcquisitions</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/LinkedinCompanyAcquisitions} obj Optional instance to populate.
     * @return {module:model/LinkedinCompanyAcquisitions} The populated <code>LinkedinCompanyAcquisitions</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new LinkedinCompanyAcquisitions();
        _Acquisition["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('acquired')) {
          obj['acquired'] = _ApiClient["default"].convertToType(data['acquired'], [_AcquiredCompany["default"]]);
        }
        if (data.hasOwnProperty('acquired_by')) {
          obj['acquired_by'] = _AcquisitionAcquiredBy["default"].constructFromObject(data['acquired_by']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>LinkedinCompanyAcquisitions</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>LinkedinCompanyAcquisitions</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['acquired']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['acquired'])) {
          throw new Error("Expected the field `acquired` to be an array in the JSON data but got " + data['acquired']);
        }
        // validate the optional field `acquired` (array)
        var _iterator = _createForOfIteratorHelper(data['acquired']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _AcquiredCompany["default"].validateJsonObject(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      // validate the optional field `acquired_by`
      if (data['acquired_by']) {
        // data not null
        _AcquisitionAcquiredBy["default"].validateJSON(data['acquired_by']);
      }
      return true;
    }
  }]);
  return LinkedinCompanyAcquisitions;
}();
/**
 * @member {Array.<module:model/AcquiredCompany>} acquired
 */
LinkedinCompanyAcquisitions.prototype['acquired'] = undefined;

/**
 * @member {module:model/AcquisitionAcquiredBy} acquired_by
 */
LinkedinCompanyAcquisitions.prototype['acquired_by'] = undefined;

// Implement Acquisition interface:
/**
 * @member {Array.<module:model/AcquiredCompany>} acquired
 */
_Acquisition["default"].prototype['acquired'] = undefined;
/**
 * @member {module:model/AcquisitionAcquiredBy} acquired_by
 */
_Acquisition["default"].prototype['acquired_by'] = undefined;
var _default = LinkedinCompanyAcquisitions;
exports["default"] = _default;
