// proxycurl-js-linkedin-profile-scraper/dist/model/LinkedinJobLocation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _JobLocation = _interopRequireDefault(require("./JobLocation"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The LinkedinJobLocation model module.
 * @module model/LinkedinJobLocation
 * @version 1.0.0
 */
var LinkedinJobLocation = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>LinkedinJobLocation</code>.
   * @alias module:model/LinkedinJobLocation
   * @implements module:model/JobLocation
   */
  function LinkedinJobLocation() {
    _classCallCheck(this, LinkedinJobLocation);
    _JobLocation["default"].initialize(this);
    LinkedinJobLocation.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(LinkedinJobLocation, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>LinkedinJobLocation</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/LinkedinJobLocation} obj Optional instance to populate.
     * @return {module:model/LinkedinJobLocation} The populated <code>LinkedinJobLocation</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new LinkedinJobLocation();
        _JobLocation["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('country')) {
          obj['country'] = _ApiClient["default"].convertToType(data['country'], 'String');
        }
        if (data.hasOwnProperty('region')) {
          obj['region'] = _ApiClient["default"].convertToType(data['region'], 'String');
        }
        if (data.hasOwnProperty('city')) {
          obj['city'] = _ApiClient["default"].convertToType(data['city'], 'String');
        }
        if (data.hasOwnProperty('postal_code')) {
          obj['postal_code'] = _ApiClient["default"].convertToType(data['postal_code'], 'String');
        }
        if (data.hasOwnProperty('latitude')) {
          obj['latitude'] = _ApiClient["default"].convertToType(data['latitude'], 'Number');
        }
        if (data.hasOwnProperty('longitude')) {
          obj['longitude'] = _ApiClient["default"].convertToType(data['longitude'], 'Number');
        }
        if (data.hasOwnProperty('street')) {
          obj['street'] = _ApiClient["default"].convertToType(data['street'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>LinkedinJobLocation</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>LinkedinJobLocation</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['country'] && !(typeof data['country'] === 'string' || data['country'] instanceof String)) {
        throw new Error("Expected the field `country` to be a primitive type in the JSON string but got " + data['country']);
      }
      // ensure the json data is a string
      if (data['region'] && !(typeof data['region'] === 'string' || data['region'] instanceof String)) {
        throw new Error("Expected the field `region` to be a primitive type in the JSON string but got " + data['region']);
      }
      // ensure the json data is a string
      if (data['city'] && !(typeof data['city'] === 'string' || data['city'] instanceof String)) {
        throw new Error("Expected the field `city` to be a primitive type in the JSON string but got " + data['city']);
      }
      // ensure the json data is a string
      if (data['postal_code'] && !(typeof data['postal_code'] === 'string' || data['postal_code'] instanceof String)) {
        throw new Error("Expected the field `postal_code` to be a primitive type in the JSON string but got " + data['postal_code']);
      }
      // ensure the json data is a string
      if (data['street'] && !(typeof data['street'] === 'string' || data['street'] instanceof String)) {
        throw new Error("Expected the field `street` to be a primitive type in the JSON string but got " + data['street']);
      }
      return true;
    }
  }]);
  return LinkedinJobLocation;
}();
/**
 * @member {String} country
 */
LinkedinJobLocation.prototype['country'] = undefined;

/**
 * @member {String} region
 */
LinkedinJobLocation.prototype['region'] = undefined;

/**
 * @member {String} city
 */
LinkedinJobLocation.prototype['city'] = undefined;

/**
 * @member {String} postal_code
 */
LinkedinJobLocation.prototype['postal_code'] = undefined;

/**
 * @member {Number} latitude
 */
LinkedinJobLocation.prototype['latitude'] = undefined;

/**
 * @member {Number} longitude
 */
LinkedinJobLocation.prototype['longitude'] = undefined;

/**
 * @member {String} street
 */
LinkedinJobLocation.prototype['street'] = undefined;

// Implement JobLocation interface:
/**
 * @member {String} country
 */
_JobLocation["default"].prototype['country'] = undefined;
/**
 * @member {String} region
 */
_JobLocation["default"].prototype['region'] = undefined;
/**
 * @member {String} city
 */
_JobLocation["default"].prototype['city'] = undefined;
/**
 * @member {String} postal_code
 */
_JobLocation["default"].prototype['postal_code'] = undefined;
/**
 * @member {Number} latitude
 */
_JobLocation["default"].prototype['latitude'] = undefined;
/**
 * @member {Number} longitude
 */
_JobLocation["default"].prototype['longitude'] = undefined;
/**
 * @member {String} street
 */
_JobLocation["default"].prototype['street'] = undefined;
var _default = LinkedinJobLocation;
exports["default"] = _default;
