// proxycurl-js-linkedin-profile-scraper/dist/model/LinkedinSchool.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _CompanyLocation = _interopRequireDefault(require("./CompanyLocation"));
var _CompanyUpdate = _interopRequireDefault(require("./CompanyUpdate"));
var _LinkedinCompanyHq = _interopRequireDefault(require("./LinkedinCompanyHq"));
var _SimilarCompany = _interopRequireDefault(require("./SimilarCompany"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The LinkedinSchool model module.
 * @module model/LinkedinSchool
 * @version 1.0.0
 */
var LinkedinSchool = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>LinkedinSchool</code>.
   * @alias module:model/LinkedinSchool
   */
  function LinkedinSchool() {
    _classCallCheck(this, LinkedinSchool);
    LinkedinSchool.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(LinkedinSchool, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>LinkedinSchool</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/LinkedinSchool} obj Optional instance to populate.
     * @return {module:model/LinkedinSchool} The populated <code>LinkedinSchool</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new LinkedinSchool();
        if (data.hasOwnProperty('linkedin_internal_id')) {
          obj['linkedin_internal_id'] = _ApiClient["default"].convertToType(data['linkedin_internal_id'], 'String');
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
        if (data.hasOwnProperty('website')) {
          obj['website'] = _ApiClient["default"].convertToType(data['website'], 'String');
        }
        if (data.hasOwnProperty('industry')) {
          obj['industry'] = _ApiClient["default"].convertToType(data['industry'], 'String');
        }
        if (data.hasOwnProperty('company_size')) {
          obj['company_size'] = _ApiClient["default"].convertToType(data['company_size'], ['Number']);
        }
        if (data.hasOwnProperty('company_size_on_linkedin')) {
          obj['company_size_on_linkedin'] = _ApiClient["default"].convertToType(data['company_size_on_linkedin'], 'Number');
        }
        if (data.hasOwnProperty('hq')) {
          obj['hq'] = _LinkedinCompanyHq["default"].constructFromObject(data['hq']);
        }
        if (data.hasOwnProperty('company_type')) {
          obj['company_type'] = _ApiClient["default"].convertToType(data['company_type'], 'String');
        }
        if (data.hasOwnProperty('founded_year')) {
          obj['founded_year'] = _ApiClient["default"].convertToType(data['founded_year'], 'Number');
        }
        if (data.hasOwnProperty('specialities')) {
          obj['specialities'] = _ApiClient["default"].convertToType(data['specialities'], ['String']);
        }
        if (data.hasOwnProperty('locations')) {
          obj['locations'] = _ApiClient["default"].convertToType(data['locations'], [_CompanyLocation["default"]]);
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('tagline')) {
          obj['tagline'] = _ApiClient["default"].convertToType(data['tagline'], 'String');
        }
        if (data.hasOwnProperty('universal_name_id')) {
          obj['universal_name_id'] = _ApiClient["default"].convertToType(data['universal_name_id'], 'String');
        }
        if (data.hasOwnProperty('profile_pic_url')) {
          obj['profile_pic_url'] = _ApiClient["default"].convertToType(data['profile_pic_url'], 'String');
        }
        if (data.hasOwnProperty('background_cover_image_url')) {
          obj['background_cover_image_url'] = _ApiClient["default"].convertToType(data['background_cover_image_url'], 'String');
        }
        if (data.hasOwnProperty('search_id')) {
          obj['search_id'] = _ApiClient["default"].convertToType(data['search_id'], 'String');
        }
        if (data.hasOwnProperty('similar_companies')) {
          obj['similar_companies'] = _ApiClient["default"].convertToType(data['similar_companies'], [_SimilarCompany["default"]]);
        }
        if (data.hasOwnProperty('updates')) {
          obj['updates'] = _ApiClient["default"].convertToType(data['updates'], [_CompanyUpdate["default"]]);
        }
        if (data.hasOwnProperty('follower_count')) {
          obj['follower_count'] = _ApiClient["default"].convertToType(data['follower_count'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>LinkedinSchool</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>LinkedinSchool</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['linkedin_internal_id'] && !(typeof data['linkedin_internal_id'] === 'string' || data['linkedin_internal_id'] instanceof String)) {
        throw new Error("Expected the field `linkedin_internal_id` to be a primitive type in the JSON string but got " + data['linkedin_internal_id']);
      }
      // ensure the json data is a string
      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      }
      // ensure the json data is a string
      if (data['website'] && !(typeof data['website'] === 'string' || data['website'] instanceof String)) {
        throw new Error("Expected the field `website` to be a primitive type in the JSON string but got " + data['website']);
      }
      // ensure the json data is a string
      if (data['industry'] && !(typeof data['industry'] === 'string' || data['industry'] instanceof String)) {
        throw new Error("Expected the field `industry` to be a primitive type in the JSON string but got " + data['industry']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['company_size'])) {
        throw new Error("Expected the field `company_size` to be an array in the JSON data but got " + data['company_size']);
      }
      // validate the optional field `hq`
      if (data['hq']) {
        // data not null
        _LinkedinCompanyHq["default"].validateJSON(data['hq']);
      }
      // ensure the json data is a string
      if (data['company_type'] && !(typeof data['company_type'] === 'string' || data['company_type'] instanceof String)) {
        throw new Error("Expected the field `company_type` to be a primitive type in the JSON string but got " + data['company_type']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['specialities'])) {
        throw new Error("Expected the field `specialities` to be an array in the JSON data but got " + data['specialities']);
      }
      if (data['locations']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['locations'])) {
          throw new Error("Expected the field `locations` to be an array in the JSON data but got " + data['locations']);
        }
        // validate the optional field `locations` (array)
        var _iterator = _createForOfIteratorHelper(data['locations']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _CompanyLocation["default"].validateJsonObject(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['tagline'] && !(typeof data['tagline'] === 'string' || data['tagline'] instanceof String)) {
        throw new Error("Expected the field `tagline` to be a primitive type in the JSON string but got " + data['tagline']);
      }
      // ensure the json data is a string
      if (data['universal_name_id'] && !(typeof data['universal_name_id'] === 'string' || data['universal_name_id'] instanceof String)) {
        throw new Error("Expected the field `universal_name_id` to be a primitive type in the JSON string but got " + data['universal_name_id']);
      }
      // ensure the json data is a string
      if (data['profile_pic_url'] && !(typeof data['profile_pic_url'] === 'string' || data['profile_pic_url'] instanceof String)) {
        throw new Error("Expected the field `profile_pic_url` to be a primitive type in the JSON string but got " + data['profile_pic_url']);
      }
      // ensure the json data is a string
      if (data['background_cover_image_url'] && !(typeof data['background_cover_image_url'] === 'string' || data['background_cover_image_url'] instanceof String)) {
        throw new Error("Expected the field `background_cover_image_url` to be a primitive type in the JSON string but got " + data['background_cover_image_url']);
      }
      // ensure the json data is a string
      if (data['search_id'] && !(typeof data['search_id'] === 'string' || data['search_id'] instanceof String)) {
        throw new Error("Expected the field `search_id` to be a primitive type in the JSON string but got " + data['search_id']);
      }
      if (data['similar_companies']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['similar_companies'])) {
          throw new Error("Expected the field `similar_companies` to be an array in the JSON data but got " + data['similar_companies']);
        }
        // validate the optional field `similar_companies` (array)
        var _iterator2 = _createForOfIteratorHelper(data['similar_companies']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var _item = _step2.value;
            _SimilarCompany["default"].validateJsonObject(_item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      if (data['updates']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['updates'])) {
          throw new Error("Expected the field `updates` to be an array in the JSON data but got " + data['updates']);
        }
        // validate the optional field `updates` (array)
        var _iterator3 = _createForOfIteratorHelper(data['updates']),
          _step3;
        try {
          for (_iterator3.s(); !(_step3 = _iterator3.n()).done;) {
            var _item2 = _step3.value;
            _CompanyUpdate["default"].validateJsonObject(_item2);
          }
        } catch (err) {
          _iterator3.e(err);
        } finally {
          _iterator3.f();
        }
        ;
      }
      return true;
    }
  }]);
  return LinkedinSchool;
}();
/**
 *          LinkedIn's Internal and immutable ID of this Company profile.                  
 * @member {String} linkedin_internal_id
 */
LinkedinSchool.prototype['linkedin_internal_id'] = undefined;

/**
 * @member {String} description
 */
LinkedinSchool.prototype['description'] = undefined;

/**
 * @member {String} website
 */
LinkedinSchool.prototype['website'] = undefined;

/**
 * @member {String} industry
 */
LinkedinSchool.prototype['industry'] = undefined;

/**
 * Listed range of company head count
 * @member {Array.<Number>} company_size
 */
LinkedinSchool.prototype['company_size'] = undefined;

/**
 * @member {Number} company_size_on_linkedin
 */
LinkedinSchool.prototype['company_size_on_linkedin'] = undefined;

/**
 * @member {module:model/LinkedinCompanyHq} hq
 */
LinkedinSchool.prototype['hq'] = undefined;

/**
 * Possible values:            `EDUCATIONAL`: Educational Institution          `GOVERNMENT_AGENCY`: Government Agency          `NON_PROFIT` : Nonprofit          `PARTNERSHIP` : Partnership          `PRIVATELY_HELD`: Privately Held          `PUBLIC_COMPANY`: Public Company          `SELF_EMPLOYED`: Self-Employed          `SELF_OWNED`: Sole Proprietorship
 * @member {String} company_type
 */
LinkedinSchool.prototype['company_type'] = undefined;

/**
 * @member {Number} founded_year
 */
LinkedinSchool.prototype['founded_year'] = undefined;

/**
 * @member {Array.<String>} specialities
 */
LinkedinSchool.prototype['specialities'] = undefined;

/**
 * @member {Array.<module:model/CompanyLocation>} locations
 */
LinkedinSchool.prototype['locations'] = undefined;

/**
 * @member {String} name
 */
LinkedinSchool.prototype['name'] = undefined;

/**
 * @member {String} tagline
 */
LinkedinSchool.prototype['tagline'] = undefined;

/**
 * @member {String} universal_name_id
 */
LinkedinSchool.prototype['universal_name_id'] = undefined;

/**
 * @member {String} profile_pic_url
 */
LinkedinSchool.prototype['profile_pic_url'] = undefined;

/**
 * @member {String} background_cover_image_url
 */
LinkedinSchool.prototype['background_cover_image_url'] = undefined;

/**
 * Useable with [Job listing endpoint](#jobs-api-jobs-listing-endpoint)
 * @member {String} search_id
 */
LinkedinSchool.prototype['search_id'] = undefined;

/**
 * @member {Array.<module:model/SimilarCompany>} similar_companies
 */
LinkedinSchool.prototype['similar_companies'] = undefined;

/**
 * @member {Array.<module:model/CompanyUpdate>} updates
 */
LinkedinSchool.prototype['updates'] = undefined;

/**
 * @member {Number} follower_count
 */
LinkedinSchool.prototype['follower_count'] = undefined;
var _default = LinkedinSchool;
exports["default"] = _default;
