// proxycurl-js-linkedin-profile-scraper/dist/model/Patent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PatentIssuedOn = _interopRequireDefault(require("./PatentIssuedOn"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The Patent model module.
 * @module model/Patent
 * @version 1.0.0
 */
var Patent = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Patent</code>.
   * @alias module:model/Patent
   */
  function Patent() {
    _classCallCheck(this, Patent);
    Patent.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Patent, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Patent</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Patent} obj Optional instance to populate.
     * @return {module:model/Patent} The populated <code>Patent</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Patent();
        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
        }
        if (data.hasOwnProperty('issuer')) {
          obj['issuer'] = _ApiClient["default"].convertToType(data['issuer'], 'String');
        }
        if (data.hasOwnProperty('issued_on')) {
          obj['issued_on'] = _PatentIssuedOn["default"].constructFromObject(data['issued_on']);
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
        if (data.hasOwnProperty('application_number')) {
          obj['application_number'] = _ApiClient["default"].convertToType(data['application_number'], 'String');
        }
        if (data.hasOwnProperty('patent_number')) {
          obj['patent_number'] = _ApiClient["default"].convertToType(data['patent_number'], 'String');
        }
        if (data.hasOwnProperty('url')) {
          obj['url'] = _ApiClient["default"].convertToType(data['url'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Patent</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Patent</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['title'] && !(typeof data['title'] === 'string' || data['title'] instanceof String)) {
        throw new Error("Expected the field `title` to be a primitive type in the JSON string but got " + data['title']);
      }
      // ensure the json data is a string
      if (data['issuer'] && !(typeof data['issuer'] === 'string' || data['issuer'] instanceof String)) {
        throw new Error("Expected the field `issuer` to be a primitive type in the JSON string but got " + data['issuer']);
      }
      // validate the optional field `issued_on`
      if (data['issued_on']) {
        // data not null
        _PatentIssuedOn["default"].validateJSON(data['issued_on']);
      }
      // ensure the json data is a string
      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      }
      // ensure the json data is a string
      if (data['application_number'] && !(typeof data['application_number'] === 'string' || data['application_number'] instanceof String)) {
        throw new Error("Expected the field `application_number` to be a primitive type in the JSON string but got " + data['application_number']);
      }
      // ensure the json data is a string
      if (data['patent_number'] && !(typeof data['patent_number'] === 'string' || data['patent_number'] instanceof String)) {
        throw new Error("Expected the field `patent_number` to be a primitive type in the JSON string but got " + data['patent_number']);
      }
      // ensure the json data is a string
      if (data['url'] && !(typeof data['url'] === 'string' || data['url'] instanceof String)) {
        throw new Error("Expected the field `url` to be a primitive type in the JSON string but got " + data['url']);
      }
      return true;
    }
  }]);
  return Patent;
}();
/**
 * Title of the patent.
 * @member {String} title
 */
Patent.prototype['title'] = undefined;

/**
 * The organisation body that issued the patent.
 * @member {String} issuer
 */
Patent.prototype['issuer'] = undefined;

/**
 * @member {module:model/PatentIssuedOn} issued_on
 */
Patent.prototype['issued_on'] = undefined;

/**
 * Description of the patent.
 * @member {String} description
 */
Patent.prototype['description'] = undefined;

/**
 * Numerical representation that identifies the patent.
 * @member {String} application_number
 */
Patent.prototype['application_number'] = undefined;

/**
 * Application number of the patent.
 * @member {String} patent_number
 */
Patent.prototype['patent_number'] = undefined;

/**
 * @member {String} url
 */
Patent.prototype['url'] = undefined;
var _default = Patent;
exports["default"] = _default;
