// proxycurl-js-linkedin-profile-scraper/dist/model/PDLEmailResult.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The PDLEmailResult model module.
 * @module model/PDLEmailResult
 * @version 1.0.0
 */
var PDLEmailResult = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PDLEmailResult</code>.
   * @alias module:model/PDLEmailResult
   */
  function PDLEmailResult() {
    _classCallCheck(this, PDLEmailResult);
    PDLEmailResult.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(PDLEmailResult, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>PDLEmailResult</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PDLEmailResult} obj Optional instance to populate.
     * @return {module:model/PDLEmailResult} The populated <code>PDLEmailResult</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PDLEmailResult();
        if (data.hasOwnProperty('emails')) {
          obj['emails'] = _ApiClient["default"].convertToType(data['emails'], ['String']);
        }
        if (data.hasOwnProperty('invalid_emails')) {
          obj['invalid_emails'] = _ApiClient["default"].convertToType(data['invalid_emails'], ['String']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>PDLEmailResult</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>PDLEmailResult</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is an array
      if (!Array.isArray(data['emails'])) {
        throw new Error("Expected the field `emails` to be an array in the JSON data but got " + data['emails']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['invalid_emails'])) {
        throw new Error("Expected the field `invalid_emails` to be an array in the JSON data but got " + data['invalid_emails']);
      }
      return true;
    }
  }]);
  return PDLEmailResult;
}();
/**
 * A list of personal emails
 * @member {Array.<String>} emails
 */
PDLEmailResult.prototype['emails'] = undefined;

/**
 * A list of invalid personal emails
 * @member {Array.<String>} invalid_emails
 */
PDLEmailResult.prototype['invalid_emails'] = undefined;
var _default = PDLEmailResult;
exports["default"] = _default;
