// proxycurl-js-linkedin-profile-scraper/dist/model/PeopleAlsoViewed.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The PeopleAlsoViewed model module.
 * @module model/PeopleAlsoViewed
 * @version 1.0.0
 */
var PeopleAlsoViewed = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PeopleAlsoViewed</code>.
   * @alias module:model/PeopleAlsoViewed
   */
  function PeopleAlsoViewed() {
    _classCallCheck(this, PeopleAlsoViewed);
    PeopleAlsoViewed.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(PeopleAlsoViewed, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>PeopleAlsoViewed</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PeopleAlsoViewed} obj Optional instance to populate.
     * @return {module:model/PeopleAlsoViewed} The populated <code>PeopleAlsoViewed</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PeopleAlsoViewed();
        if (data.hasOwnProperty('link')) {
          obj['link'] = _ApiClient["default"].convertToType(data['link'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('summary')) {
          obj['summary'] = _ApiClient["default"].convertToType(data['summary'], 'String');
        }
        if (data.hasOwnProperty('location')) {
          obj['location'] = _ApiClient["default"].convertToType(data['location'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>PeopleAlsoViewed</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>PeopleAlsoViewed</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['link'] && !(typeof data['link'] === 'string' || data['link'] instanceof String)) {
        throw new Error("Expected the field `link` to be a primitive type in the JSON string but got " + data['link']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['summary'] && !(typeof data['summary'] === 'string' || data['summary'] instanceof String)) {
        throw new Error("Expected the field `summary` to be a primitive type in the JSON string but got " + data['summary']);
      }
      // ensure the json data is a string
      if (data['location'] && !(typeof data['location'] === 'string' || data['location'] instanceof String)) {
        throw new Error("Expected the field `location` to be a primitive type in the JSON string but got " + data['location']);
      }
      return true;
    }
  }]);
  return PeopleAlsoViewed;
}();
/**
 *          URL of the profile.          Useable with [Person profile endpoint](#people-api-person-profile-endpoint)         
 * @member {String} link
 */
PeopleAlsoViewed.prototype['link'] = undefined;

/**
 * @member {String} name
 */
PeopleAlsoViewed.prototype['name'] = undefined;

/**
 * @member {String} summary
 */
PeopleAlsoViewed.prototype['summary'] = undefined;

/**
 * @member {String} location
 */
PeopleAlsoViewed.prototype['location'] = undefined;
var _default = PeopleAlsoViewed;
exports["default"] = _default;
