// proxycurl-js-linkedin-profile-scraper/dist/model/Person.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The Person model module.
 * @module model/Person
 * @version 1.0.0
 */
var Person = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Person</code>.
   * @alias module:model/Person
   */
  function Person() {
    _classCallCheck(this, Person);
    Person.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Person, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Person</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Person} obj Optional instance to populate.
     * @return {module:model/Person} The populated <code>Person</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Person();
        if (data.hasOwnProperty('public_identifier')) {
          obj['public_identifier'] = _ApiClient["default"].convertToType(data['public_identifier'], 'String');
        }
        if (data.hasOwnProperty('profile_pic_url')) {
          obj['profile_pic_url'] = _ApiClient["default"].convertToType(data['profile_pic_url'], 'String');
        }
        if (data.hasOwnProperty('background_cover_image_url')) {
          obj['background_cover_image_url'] = _ApiClient["default"].convertToType(data['background_cover_image_url'], 'String');
        }
        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
        }
        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
        }
        if (data.hasOwnProperty('full_name')) {
          obj['full_name'] = _ApiClient["default"].convertToType(data['full_name'], 'String');
        }
        if (data.hasOwnProperty('occupation')) {
          obj['occupation'] = _ApiClient["default"].convertToType(data['occupation'], 'String');
        }
        if (data.hasOwnProperty('headline')) {
          obj['headline'] = _ApiClient["default"].convertToType(data['headline'], 'String');
        }
        if (data.hasOwnProperty('summary')) {
          obj['summary'] = _ApiClient["default"].convertToType(data['summary'], 'String');
        }
        if (data.hasOwnProperty('country')) {
          obj['country'] = _ApiClient["default"].convertToType(data['country'], 'String');
        }
        if (data.hasOwnProperty('country_full_name')) {
          obj['country_full_name'] = _ApiClient["default"].convertToType(data['country_full_name'], 'String');
        }
        if (data.hasOwnProperty('city')) {
          obj['city'] = _ApiClient["default"].convertToType(data['city'], 'String');
        }
        if (data.hasOwnProperty('state')) {
          obj['state'] = _ApiClient["default"].convertToType(data['state'], 'String');
        }
        if (data.hasOwnProperty('experiences')) {
          obj['experiences'] = _ApiClient["default"].convertToType(data['experiences'], Array);
        }
        if (data.hasOwnProperty('education')) {
          obj['education'] = _ApiClient["default"].convertToType(data['education'], Array);
        }
        if (data.hasOwnProperty('languages')) {
          obj['languages'] = _ApiClient["default"].convertToType(data['languages'], Array);
        }
        if (data.hasOwnProperty('accomplishment_organisations')) {
          obj['accomplishment_organisations'] = _ApiClient["default"].convertToType(data['accomplishment_organisations'], Array);
        }
        if (data.hasOwnProperty('accomplishment_publications')) {
          obj['accomplishment_publications'] = _ApiClient["default"].convertToType(data['accomplishment_publications'], Array);
        }
        if (data.hasOwnProperty('accomplishment_honors_awards')) {
          obj['accomplishment_honors_awards'] = _ApiClient["default"].convertToType(data['accomplishment_honors_awards'], Array);
        }
        if (data.hasOwnProperty('accomplishment_patents')) {
          obj['accomplishment_patents'] = _ApiClient["default"].convertToType(data['accomplishment_patents'], Array);
        }
        if (data.hasOwnProperty('accomplishment_courses')) {
          obj['accomplishment_courses'] = _ApiClient["default"].convertToType(data['accomplishment_courses'], Array);
        }
        if (data.hasOwnProperty('accomplishment_projects')) {
          obj['accomplishment_projects'] = _ApiClient["default"].convertToType(data['accomplishment_projects'], Array);
        }
        if (data.hasOwnProperty('accomplishment_test_scores')) {
          obj['accomplishment_test_scores'] = _ApiClient["default"].convertToType(data['accomplishment_test_scores'], Array);
        }
        if (data.hasOwnProperty('volunteer_work')) {
          obj['volunteer_work'] = _ApiClient["default"].convertToType(data['volunteer_work'], Array);
        }
        if (data.hasOwnProperty('certifications')) {
          obj['certifications'] = _ApiClient["default"].convertToType(data['certifications'], Array);
        }
        if (data.hasOwnProperty('connections')) {
          obj['connections'] = _ApiClient["default"].convertToType(data['connections'], 'Number');
        }
        if (data.hasOwnProperty('people_also_viewed')) {
          obj['people_also_viewed'] = _ApiClient["default"].convertToType(data['people_also_viewed'], Array);
        }
        if (data.hasOwnProperty('recommendations')) {
          obj['recommendations'] = _ApiClient["default"].convertToType(data['recommendations'], Array);
        }
        if (data.hasOwnProperty('activities')) {
          obj['activities'] = _ApiClient["default"].convertToType(data['activities'], Array);
        }
        if (data.hasOwnProperty('similarly_named_profiles')) {
          obj['similarly_named_profiles'] = _ApiClient["default"].convertToType(data['similarly_named_profiles'], Array);
        }
        if (data.hasOwnProperty('articles')) {
          obj['articles'] = _ApiClient["default"].convertToType(data['articles'], Array);
        }
        if (data.hasOwnProperty('groups')) {
          obj['groups'] = _ApiClient["default"].convertToType(data['groups'], Array);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Person</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Person</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['public_identifier'] && !(typeof data['public_identifier'] === 'string' || data['public_identifier'] instanceof String)) {
        throw new Error("Expected the field `public_identifier` to be a primitive type in the JSON string but got " + data['public_identifier']);
      }
      // ensure the json data is a string
      if (data['profile_pic_url'] && !(typeof data['profile_pic_url'] === 'string' || data['profile_pic_url'] instanceof String)) {
        throw new Error("Expected the field `profile_pic_url` to be a primitive type in the JSON string but got " + data['profile_pic_url']);
      }
      // ensure the json data is a string
      if (data['background_cover_image_url'] && !(typeof data['background_cover_image_url'] === 'string' || data['background_cover_image_url'] instanceof String)) {
        throw new Error("Expected the field `background_cover_image_url` to be a primitive type in the JSON string but got " + data['background_cover_image_url']);
      }
      // ensure the json data is a string
      if (data['first_name'] && !(typeof data['first_name'] === 'string' || data['first_name'] instanceof String)) {
        throw new Error("Expected the field `first_name` to be a primitive type in the JSON string but got " + data['first_name']);
      }
      // ensure the json data is a string
      if (data['last_name'] && !(typeof data['last_name'] === 'string' || data['last_name'] instanceof String)) {
        throw new Error("Expected the field `last_name` to be a primitive type in the JSON string but got " + data['last_name']);
      }
      // ensure the json data is a string
      if (data['full_name'] && !(typeof data['full_name'] === 'string' || data['full_name'] instanceof String)) {
        throw new Error("Expected the field `full_name` to be a primitive type in the JSON string but got " + data['full_name']);
      }
      // ensure the json data is a string
      if (data['occupation'] && !(typeof data['occupation'] === 'string' || data['occupation'] instanceof String)) {
        throw new Error("Expected the field `occupation` to be a primitive type in the JSON string but got " + data['occupation']);
      }
      // ensure the json data is a string
      if (data['headline'] && !(typeof data['headline'] === 'string' || data['headline'] instanceof String)) {
        throw new Error("Expected the field `headline` to be a primitive type in the JSON string but got " + data['headline']);
      }
      // ensure the json data is a string
      if (data['summary'] && !(typeof data['summary'] === 'string' || data['summary'] instanceof String)) {
        throw new Error("Expected the field `summary` to be a primitive type in the JSON string but got " + data['summary']);
      }
      // ensure the json data is a string
      if (data['country'] && !(typeof data['country'] === 'string' || data['country'] instanceof String)) {
        throw new Error("Expected the field `country` to be a primitive type in the JSON string but got " + data['country']);
      }
      // ensure the json data is a string
      if (data['country_full_name'] && !(typeof data['country_full_name'] === 'string' || data['country_full_name'] instanceof String)) {
        throw new Error("Expected the field `country_full_name` to be a primitive type in the JSON string but got " + data['country_full_name']);
      }
      // ensure the json data is a string
      if (data['city'] && !(typeof data['city'] === 'string' || data['city'] instanceof String)) {
        throw new Error("Expected the field `city` to be a primitive type in the JSON string but got " + data['city']);
      }
      // ensure the json data is a string
      if (data['state'] && !(typeof data['state'] === 'string' || data['state'] instanceof String)) {
        throw new Error("Expected the field `state` to be a primitive type in the JSON string but got " + data['state']);
      }
      return true;
    }
  }]);
  return Person;
}();
/**
 *                  The vanity identifier of the public LinkedIn profile.                 The vanity identifier comes after the `/in/` part of the LinkedIn Profile URL                 in the following format: `https://www.linkedin.com/in/<public_identifier>`             
 * @member {String} public_identifier
 */
Person.prototype['public_identifier'] = undefined;

/**
 *                  A temporary link to the user's profile picture that is valid for 30 minutes.                  The temporal nature of the link is by design to prevent having Proxycurl be the mirror for the images.                 The developer is expected to handle these images by downloading the image and re-hosting the image.                 See [this post](https://nubela.co/blog/why-is-the-api-returning-s3-links-for-profile-pictures-scraped-from-linkedin-profiles/) for context.                 Some profile pictures might be of the standard LinkedIn's profile picture placeholder. It is so because. See [this post](https://nubela.co/blog/why-do-most-linkedin-profiles-fetched-via-the-person-profile-endpoint-return-a-placeholder-profile-picture/) for context.             
 * @member {String} profile_pic_url
 */
Person.prototype['profile_pic_url'] = undefined;

/**
 *                  A temporary link to the user's background cover picture                 that is valid for 30 minutes.                 The temporal nature of the link is by design to prevent                 having Proxycurl be the mirror for the images.                 The developer is expected to handle these images                  by downloading the image and re-hosting the image.                  See [this post](https://nubela.co/blog/why-is-the-api-returning-s3-links-for-profile-pictures-scraped-from-linkedin-profiles/) for context.             
 * @member {String} background_cover_image_url
 */
Person.prototype['background_cover_image_url'] = undefined;

/**
 * First name of the user.
 * @member {String} first_name
 */
Person.prototype['first_name'] = undefined;

/**
 * Last name of the user.
 * @member {String} last_name
 */
Person.prototype['last_name'] = undefined;

/**
 *                  Full name of the user (`first_name` + `last_name`)             
 * @member {String} full_name
 */
Person.prototype['full_name'] = undefined;

/**
 *                  The title and company name of the user's current employment.             
 * @member {String} occupation
 */
Person.prototype['occupation'] = undefined;

/**
 *                  The tagline written by the user for his profile.             
 * @member {String} headline
 */
Person.prototype['headline'] = undefined;

/**
 *                  A blurb (longer than the tagline) written by the user for his profile.             
 * @member {String} summary
 */
Person.prototype['summary'] = undefined;

/**
 *                  The user's country of residence depicted by                 a 2-letter country code (ISO 3166-1 alpha-2).             
 * @member {String} country
 */
Person.prototype['country'] = undefined;

/**
 * The user's country of residence, in English words.
 * @member {String} country_full_name
 */
Person.prototype['country_full_name'] = undefined;

/**
 * The city that the user is living at.
 * @member {String} city
 */
Person.prototype['city'] = undefined;

/**
 * The state that the user is living at.
 * @member {String} state
 */
Person.prototype['state'] = undefined;

/**
 * The user's list of historic work experiences.
 * @member {Array} experiences
 */
Person.prototype['experiences'] = undefined;

/**
 * The user's list of education background.
 * @member {Array} education
 */
Person.prototype['education'] = undefined;

/**
 *                  A list of languages that the user claims to be familiar with,                 and has added to his/her profile.                 Do note that we do not have the proficiency level as                 that data point is not available on a public LinkedIn profile.             
 * @member {Array} languages
 */
Person.prototype['languages'] = undefined;

/**
 *                  List of noteworthy organizations that this user is part of.             
 * @member {Array} accomplishment_organisations
 */
Person.prototype['accomplishment_organisations'] = undefined;

/**
 *                  List of noteworthy publications that this user has partook in.             
 * @member {Array} accomplishment_publications
 */
Person.prototype['accomplishment_publications'] = undefined;

/**
 *                  List of noteworthy honours and awards that this user has won.             
 * @member {Array} accomplishment_honors_awards
 */
Person.prototype['accomplishment_honors_awards'] = undefined;

/**
 * List of noteworthy patents won by this user.
 * @member {Array} accomplishment_patents
 */
Person.prototype['accomplishment_patents'] = undefined;

/**
 * List of noteworthy courses partook by this user.
 * @member {Array} accomplishment_courses
 */
Person.prototype['accomplishment_courses'] = undefined;

/**
 *                  List of noteworthy projects undertaken by this user.             
 * @member {Array} accomplishment_projects
 */
Person.prototype['accomplishment_projects'] = undefined;

/**
 *                  List of noteworthy test scores accomplished by this user.             
 * @member {Array} accomplishment_test_scores
 */
Person.prototype['accomplishment_test_scores'] = undefined;

/**
 * List of historic volunteer work experiences.
 * @member {Array} volunteer_work
 */
Person.prototype['volunteer_work'] = undefined;

/**
 *                  List of noteworthy certifications accomplished by this user.             
 * @member {Array} certifications
 */
Person.prototype['certifications'] = undefined;

/**
 * Total *count* of LinkedIn connections.
 * @member {Number} connections
 */
Person.prototype['connections'] = undefined;

/**
 *                  A list of other LinkedIn profiles closely related to this user.             
 * @member {Array} people_also_viewed
 */
Person.prototype['people_also_viewed'] = undefined;

/**
 *                  List of recommendations made by other users about this profile.             
 * @member {Array} recommendations
 */
Person.prototype['recommendations'] = undefined;

/**
 * A list of LinkedIn status activities.
 * @member {Array} activities
 */
Person.prototype['activities'] = undefined;

/**
 *                  A list of other LinkedIn profiles with similar names.             
 * @member {Array} similarly_named_profiles
 */
Person.prototype['similarly_named_profiles'] = undefined;

/**
 *                  A list of content-based articles posted by this user.             
 * @member {Array} articles
 */
Person.prototype['articles'] = undefined;

/**
 *                  A list of LinkedIn groups that this user is a part of.\",             
 * @member {Array} groups
 */
Person.prototype['groups'] = undefined;
var _default = Person;
exports["default"] = _default;
