// proxycurl-js-linkedin-profile-scraper/dist/model/PersonExtra.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The PersonExtra model module.
 * @module model/PersonExtra
 * @version 1.0.0
 */
var PersonExtra = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonExtra</code>.
   * @alias module:model/PersonExtra
   */
  function PersonExtra() {
    _classCallCheck(this, PersonExtra);
    PersonExtra.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(PersonExtra, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>PersonExtra</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonExtra} obj Optional instance to populate.
     * @return {module:model/PersonExtra} The populated <code>PersonExtra</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonExtra();
        if (data.hasOwnProperty('github_profile_id')) {
          obj['github_profile_id'] = _ApiClient["default"].convertToType(data['github_profile_id'], 'String');
        }
        if (data.hasOwnProperty('facebook_profile_id')) {
          obj['facebook_profile_id'] = _ApiClient["default"].convertToType(data['facebook_profile_id'], 'String');
        }
        if (data.hasOwnProperty('twitter_profile_id')) {
          obj['twitter_profile_id'] = _ApiClient["default"].convertToType(data['twitter_profile_id'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>PersonExtra</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>PersonExtra</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['github_profile_id'] && !(typeof data['github_profile_id'] === 'string' || data['github_profile_id'] instanceof String)) {
        throw new Error("Expected the field `github_profile_id` to be a primitive type in the JSON string but got " + data['github_profile_id']);
      }
      // ensure the json data is a string
      if (data['facebook_profile_id'] && !(typeof data['facebook_profile_id'] === 'string' || data['facebook_profile_id'] instanceof String)) {
        throw new Error("Expected the field `facebook_profile_id` to be a primitive type in the JSON string but got " + data['facebook_profile_id']);
      }
      // ensure the json data is a string
      if (data['twitter_profile_id'] && !(typeof data['twitter_profile_id'] === 'string' || data['twitter_profile_id'] instanceof String)) {
        throw new Error("Expected the field `twitter_profile_id` to be a primitive type in the JSON string but got " + data['twitter_profile_id']);
      }
      return true;
    }
  }]);
  return PersonExtra;
}();
/**
 * This profile's Github account.
 * @member {String} github_profile_id
 */
PersonExtra.prototype['github_profile_id'] = undefined;

/**
 * This profile's Facebook account.
 * @member {String} facebook_profile_id
 */
PersonExtra.prototype['facebook_profile_id'] = undefined;

/**
 * This profile's twitter account.
 * @member {String} twitter_profile_id
 */
PersonExtra.prototype['twitter_profile_id'] = undefined;
var _default = PersonExtra;
exports["default"] = _default;
