// proxycurl-js-linkedin-profile-scraper/dist/model/PersonEndpointResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AccomplishmentOrg = _interopRequireDefault(require("./AccomplishmentOrg"));
var _Activity = _interopRequireDefault(require("./Activity"));
var _Article = _interopRequireDefault(require("./Article"));
var _Certification = _interopRequireDefault(require("./Certification"));
var _Course = _interopRequireDefault(require("./Course"));
var _Education = _interopRequireDefault(require("./Education"));
var _Experience = _interopRequireDefault(require("./Experience"));
var _HonourAward = _interopRequireDefault(require("./HonourAward"));
var _Patent = _interopRequireDefault(require("./Patent"));
var _PeopleAlsoViewed = _interopRequireDefault(require("./PeopleAlsoViewed"));
var _PersonEndpointResponseBirthDate = _interopRequireDefault(require("./PersonEndpointResponseBirthDate"));
var _PersonEndpointResponseExtra = _interopRequireDefault(require("./PersonEndpointResponseExtra"));
var _PersonEndpointResponseInferredSalary = _interopRequireDefault(require("./PersonEndpointResponseInferredSalary"));
var _PersonGroup = _interopRequireDefault(require("./PersonGroup"));
var _Project = _interopRequireDefault(require("./Project"));
var _Publication = _interopRequireDefault(require("./Publication"));
var _SimilarProfile = _interopRequireDefault(require("./SimilarProfile"));
var _TestScore = _interopRequireDefault(require("./TestScore"));
var _VolunteeringExperience = _interopRequireDefault(require("./VolunteeringExperience"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The PersonEndpointResponse model module.
 * @module model/PersonEndpointResponse
 * @version 1.0.0
 */
var PersonEndpointResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonEndpointResponse</code>.
   * @alias module:model/PersonEndpointResponse
   */
  function PersonEndpointResponse() {
    _classCallCheck(this, PersonEndpointResponse);
    PersonEndpointResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(PersonEndpointResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>PersonEndpointResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonEndpointResponse} obj Optional instance to populate.
     * @return {module:model/PersonEndpointResponse} The populated <code>PersonEndpointResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonEndpointResponse();
        if (data.hasOwnProperty('public_identifier')) {
          obj['public_identifier'] = _ApiClient["default"].convertToType(data['public_identifier'], 'String');
        }
        if (data.hasOwnProperty('profile_pic_url')) {
          obj['profile_pic_url'] = _ApiClient["default"].convertToType(data['profile_pic_url'], 'String');
        }
        if (data.hasOwnProperty('background_cover_image_url')) {
          obj['background_cover_image_url'] = _ApiClient["default"].convertToType(data['background_cover_image_url'], 'String');
        }
        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
        }
        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
        }
        if (data.hasOwnProperty('full_name')) {
          obj['full_name'] = _ApiClient["default"].convertToType(data['full_name'], 'String');
        }
        if (data.hasOwnProperty('occupation')) {
          obj['occupation'] = _ApiClient["default"].convertToType(data['occupation'], 'String');
        }
        if (data.hasOwnProperty('headline')) {
          obj['headline'] = _ApiClient["default"].convertToType(data['headline'], 'String');
        }
        if (data.hasOwnProperty('summary')) {
          obj['summary'] = _ApiClient["default"].convertToType(data['summary'], 'String');
        }
        if (data.hasOwnProperty('country')) {
          obj['country'] = _ApiClient["default"].convertToType(data['country'], 'String');
        }
        if (data.hasOwnProperty('country_full_name')) {
          obj['country_full_name'] = _ApiClient["default"].convertToType(data['country_full_name'], 'String');
        }
        if (data.hasOwnProperty('city')) {
          obj['city'] = _ApiClient["default"].convertToType(data['city'], 'String');
        }
        if (data.hasOwnProperty('state')) {
          obj['state'] = _ApiClient["default"].convertToType(data['state'], 'String');
        }
        if (data.hasOwnProperty('experiences')) {
          obj['experiences'] = _ApiClient["default"].convertToType(data['experiences'], [_Experience["default"]]);
        }
        if (data.hasOwnProperty('education')) {
          obj['education'] = _ApiClient["default"].convertToType(data['education'], [_Education["default"]]);
        }
        if (data.hasOwnProperty('languages')) {
          obj['languages'] = _ApiClient["default"].convertToType(data['languages'], ['String']);
        }
        if (data.hasOwnProperty('accomplishment_organisations')) {
          obj['accomplishment_organisations'] = _ApiClient["default"].convertToType(data['accomplishment_organisations'], [_AccomplishmentOrg["default"]]);
        }
        if (data.hasOwnProperty('accomplishment_publications')) {
          obj['accomplishment_publications'] = _ApiClient["default"].convertToType(data['accomplishment_publications'], [_Publication["default"]]);
        }
        if (data.hasOwnProperty('accomplishment_honors_awards')) {
          obj['accomplishment_honors_awards'] = _ApiClient["default"].convertToType(data['accomplishment_honors_awards'], [_HonourAward["default"]]);
        }
        if (data.hasOwnProperty('accomplishment_patents')) {
          obj['accomplishment_patents'] = _ApiClient["default"].convertToType(data['accomplishment_patents'], [_Patent["default"]]);
        }
        if (data.hasOwnProperty('accomplishment_courses')) {
          obj['accomplishment_courses'] = _ApiClient["default"].convertToType(data['accomplishment_courses'], [_Course["default"]]);
        }
        if (data.hasOwnProperty('accomplishment_projects')) {
          obj['accomplishment_projects'] = _ApiClient["default"].convertToType(data['accomplishment_projects'], [_Project["default"]]);
        }
        if (data.hasOwnProperty('accomplishment_test_scores')) {
          obj['accomplishment_test_scores'] = _ApiClient["default"].convertToType(data['accomplishment_test_scores'], [_TestScore["default"]]);
        }
        if (data.hasOwnProperty('volunteer_work')) {
          obj['volunteer_work'] = _ApiClient["default"].convertToType(data['volunteer_work'], [_VolunteeringExperience["default"]]);
        }
        if (data.hasOwnProperty('certifications')) {
          obj['certifications'] = _ApiClient["default"].convertToType(data['certifications'], [_Certification["default"]]);
        }
        if (data.hasOwnProperty('connections')) {
          obj['connections'] = _ApiClient["default"].convertToType(data['connections'], 'Number');
        }
        if (data.hasOwnProperty('people_also_viewed')) {
          obj['people_also_viewed'] = _ApiClient["default"].convertToType(data['people_also_viewed'], [_PeopleAlsoViewed["default"]]);
        }
        if (data.hasOwnProperty('recommendations')) {
          obj['recommendations'] = _ApiClient["default"].convertToType(data['recommendations'], ['String']);
        }
        if (data.hasOwnProperty('activities')) {
          obj['activities'] = _ApiClient["default"].convertToType(data['activities'], [_Activity["default"]]);
        }
        if (data.hasOwnProperty('similarly_named_profiles')) {
          obj['similarly_named_profiles'] = _ApiClient["default"].convertToType(data['similarly_named_profiles'], [_SimilarProfile["default"]]);
        }
        if (data.hasOwnProperty('articles')) {
          obj['articles'] = _ApiClient["default"].convertToType(data['articles'], [_Article["default"]]);
        }
        if (data.hasOwnProperty('groups')) {
          obj['groups'] = _ApiClient["default"].convertToType(data['groups'], [_PersonGroup["default"]]);
        }
        if (data.hasOwnProperty('skills')) {
          obj['skills'] = _ApiClient["default"].convertToType(data['skills'], ['String']);
        }
        if (data.hasOwnProperty('inferred_salary')) {
          obj['inferred_salary'] = _PersonEndpointResponseInferredSalary["default"].constructFromObject(data['inferred_salary']);
        }
        if (data.hasOwnProperty('gender')) {
          obj['gender'] = _ApiClient["default"].convertToType(data['gender'], 'String');
        }
        if (data.hasOwnProperty('birth_date')) {
          obj['birth_date'] = _PersonEndpointResponseBirthDate["default"].constructFromObject(data['birth_date']);
        }
        if (data.hasOwnProperty('industry')) {
          obj['industry'] = _ApiClient["default"].convertToType(data['industry'], 'String');
        }
        if (data.hasOwnProperty('interests')) {
          obj['interests'] = _ApiClient["default"].convertToType(data['interests'], ['String']);
        }
        if (data.hasOwnProperty('extra')) {
          obj['extra'] = _PersonEndpointResponseExtra["default"].constructFromObject(data['extra']);
        }
        if (data.hasOwnProperty('personal_emails')) {
          obj['personal_emails'] = _ApiClient["default"].convertToType(data['personal_emails'], ['String']);
        }
        if (data.hasOwnProperty('personal_numbers')) {
          obj['personal_numbers'] = _ApiClient["default"].convertToType(data['personal_numbers'], ['String']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>PersonEndpointResponse</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>PersonEndpointResponse</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['public_identifier'] && !(typeof data['public_identifier'] === 'string' || data['public_identifier'] instanceof String)) {
        throw new Error("Expected the field `public_identifier` to be a primitive type in the JSON string but got " + data['public_identifier']);
      }
      // ensure the json data is a string
      if (data['profile_pic_url'] && !(typeof data['profile_pic_url'] === 'string' || data['profile_pic_url'] instanceof String)) {
        throw new Error("Expected the field `profile_pic_url` to be a primitive type in the JSON string but got " + data['profile_pic_url']);
      }
      // ensure the json data is a string
      if (data['background_cover_image_url'] && !(typeof data['background_cover_image_url'] === 'string' || data['background_cover_image_url'] instanceof String)) {
        throw new Error("Expected the field `background_cover_image_url` to be a primitive type in the JSON string but got " + data['background_cover_image_url']);
      }
      // ensure the json data is a string
      if (data['first_name'] && !(typeof data['first_name'] === 'string' || data['first_name'] instanceof String)) {
        throw new Error("Expected the field `first_name` to be a primitive type in the JSON string but got " + data['first_name']);
      }
      // ensure the json data is a string
      if (data['last_name'] && !(typeof data['last_name'] === 'string' || data['last_name'] instanceof String)) {
        throw new Error("Expected the field `last_name` to be a primitive type in the JSON string but got " + data['last_name']);
      }
      // ensure the json data is a string
      if (data['full_name'] && !(typeof data['full_name'] === 'string' || data['full_name'] instanceof String)) {
        throw new Error("Expected the field `full_name` to be a primitive type in the JSON string but got " + data['full_name']);
      }
      // ensure the json data is a string
      if (data['occupation'] && !(typeof data['occupation'] === 'string' || data['occupation'] instanceof String)) {
        throw new Error("Expected the field `occupation` to be a primitive type in the JSON string but got " + data['occupation']);
      }
      // ensure the json data is a string
      if (data['headline'] && !(typeof data['headline'] === 'string' || data['headline'] instanceof String)) {
        throw new Error("Expected the field `headline` to be a primitive type in the JSON string but got " + data['headline']);
      }
      // ensure the json data is a string
      if (data['summary'] && !(typeof data['summary'] === 'string' || data['summary'] instanceof String)) {
        throw new Error("Expected the field `summary` to be a primitive type in the JSON string but got " + data['summary']);
      }
      // ensure the json data is a string
      if (data['country'] && !(typeof data['country'] === 'string' || data['country'] instanceof String)) {
        throw new Error("Expected the field `country` to be a primitive type in the JSON string but got " + data['country']);
      }
      // ensure the json data is a string
      if (data['country_full_name'] && !(typeof data['country_full_name'] === 'string' || data['country_full_name'] instanceof String)) {
        throw new Error("Expected the field `country_full_name` to be a primitive type in the JSON string but got " + data['country_full_name']);
      }
      // ensure the json data is a string
      if (data['city'] && !(typeof data['city'] === 'string' || data['city'] instanceof String)) {
        throw new Error("Expected the field `city` to be a primitive type in the JSON string but got " + data['city']);
      }
      // ensure the json data is a string
      if (data['state'] && !(typeof data['state'] === 'string' || data['state'] instanceof String)) {
        throw new Error("Expected the field `state` to be a primitive type in the JSON string but got " + data['state']);
      }
      if (data['experiences']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['experiences'])) {
          throw new Error("Expected the field `experiences` to be an array in the JSON data but got " + data['experiences']);
        }
        // validate the optional field `experiences` (array)
        var _iterator = _createForOfIteratorHelper(data['experiences']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _Experience["default"].validateJsonObject(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      if (data['education']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['education'])) {
          throw new Error("Expected the field `education` to be an array in the JSON data but got " + data['education']);
        }
        // validate the optional field `education` (array)
        var _iterator2 = _createForOfIteratorHelper(data['education']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var _item = _step2.value;
            _Education["default"].validateJsonObject(_item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      // ensure the json data is an array
      if (!Array.isArray(data['languages'])) {
        throw new Error("Expected the field `languages` to be an array in the JSON data but got " + data['languages']);
      }
      if (data['accomplishment_organisations']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['accomplishment_organisations'])) {
          throw new Error("Expected the field `accomplishment_organisations` to be an array in the JSON data but got " + data['accomplishment_organisations']);
        }
        // validate the optional field `accomplishment_organisations` (array)
        var _iterator3 = _createForOfIteratorHelper(data['accomplishment_organisations']),
          _step3;
        try {
          for (_iterator3.s(); !(_step3 = _iterator3.n()).done;) {
            var _item2 = _step3.value;
            _AccomplishmentOrg["default"].validateJsonObject(_item2);
          }
        } catch (err) {
          _iterator3.e(err);
        } finally {
          _iterator3.f();
        }
        ;
      }
      if (data['accomplishment_publications']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['accomplishment_publications'])) {
          throw new Error("Expected the field `accomplishment_publications` to be an array in the JSON data but got " + data['accomplishment_publications']);
        }
        // validate the optional field `accomplishment_publications` (array)
        var _iterator4 = _createForOfIteratorHelper(data['accomplishment_publications']),
          _step4;
        try {
          for (_iterator4.s(); !(_step4 = _iterator4.n()).done;) {
            var _item3 = _step4.value;
            _Publication["default"].validateJsonObject(_item3);
          }
        } catch (err) {
          _iterator4.e(err);
        } finally {
          _iterator4.f();
        }
        ;
      }
      if (data['accomplishment_honors_awards']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['accomplishment_honors_awards'])) {
          throw new Error("Expected the field `accomplishment_honors_awards` to be an array in the JSON data but got " + data['accomplishment_honors_awards']);
        }
        // validate the optional field `accomplishment_honors_awards` (array)
        var _iterator5 = _createForOfIteratorHelper(data['accomplishment_honors_awards']),
          _step5;
        try {
          for (_iterator5.s(); !(_step5 = _iterator5.n()).done;) {
            var _item4 = _step5.value;
            _HonourAward["default"].validateJsonObject(_item4);
          }
        } catch (err) {
          _iterator5.e(err);
        } finally {
          _iterator5.f();
        }
        ;
      }
      if (data['accomplishment_patents']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['accomplishment_patents'])) {
          throw new Error("Expected the field `accomplishment_patents` to be an array in the JSON data but got " + data['accomplishment_patents']);
        }
        // validate the optional field `accomplishment_patents` (array)
        var _iterator6 = _createForOfIteratorHelper(data['accomplishment_patents']),
          _step6;
        try {
          for (_iterator6.s(); !(_step6 = _iterator6.n()).done;) {
            var _item5 = _step6.value;
            _Patent["default"].validateJsonObject(_item5);
          }
        } catch (err) {
          _iterator6.e(err);
        } finally {
          _iterator6.f();
        }
        ;
      }
      if (data['accomplishment_courses']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['accomplishment_courses'])) {
          throw new Error("Expected the field `accomplishment_courses` to be an array in the JSON data but got " + data['accomplishment_courses']);
        }
        // validate the optional field `accomplishment_courses` (array)
        var _iterator7 = _createForOfIteratorHelper(data['accomplishment_courses']),
          _step7;
        try {
          for (_iterator7.s(); !(_step7 = _iterator7.n()).done;) {
            var _item6 = _step7.value;
            _Course["default"].validateJsonObject(_item6);
          }
        } catch (err) {
          _iterator7.e(err);
        } finally {
          _iterator7.f();
        }
        ;
      }
      if (data['accomplishment_projects']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['accomplishment_projects'])) {
          throw new Error("Expected the field `accomplishment_projects` to be an array in the JSON data but got " + data['accomplishment_projects']);
        }
        // validate the optional field `accomplishment_projects` (array)
        var _iterator8 = _createForOfIteratorHelper(data['accomplishment_projects']),
          _step8;
        try {
          for (_iterator8.s(); !(_step8 = _iterator8.n()).done;) {
            var _item7 = _step8.value;
            _Project["default"].validateJsonObject(_item7);
          }
        } catch (err) {
          _iterator8.e(err);
        } finally {
          _iterator8.f();
        }
        ;
      }
      if (data['accomplishment_test_scores']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['accomplishment_test_scores'])) {
          throw new Error("Expected the field `accomplishment_test_scores` to be an array in the JSON data but got " + data['accomplishment_test_scores']);
        }
        // validate the optional field `accomplishment_test_scores` (array)
        var _iterator9 = _createForOfIteratorHelper(data['accomplishment_test_scores']),
          _step9;
        try {
          for (_iterator9.s(); !(_step9 = _iterator9.n()).done;) {
            var _item8 = _step9.value;
            _TestScore["default"].validateJsonObject(_item8);
          }
        } catch (err) {
          _iterator9.e(err);
        } finally {
          _iterator9.f();
        }
        ;
      }
      if (data['volunteer_work']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['volunteer_work'])) {
          throw new Error("Expected the field `volunteer_work` to be an array in the JSON data but got " + data['volunteer_work']);
        }
        // validate the optional field `volunteer_work` (array)
        var _iterator10 = _createForOfIteratorHelper(data['volunteer_work']),
          _step10;
        try {
          for (_iterator10.s(); !(_step10 = _iterator10.n()).done;) {
            var _item9 = _step10.value;
            _VolunteeringExperience["default"].validateJsonObject(_item9);
          }
        } catch (err) {
          _iterator10.e(err);
        } finally {
          _iterator10.f();
        }
        ;
      }
      if (data['certifications']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['certifications'])) {
          throw new Error("Expected the field `certifications` to be an array in the JSON data but got " + data['certifications']);
        }
        // validate the optional field `certifications` (array)
        var _iterator11 = _createForOfIteratorHelper(data['certifications']),
          _step11;
        try {
          for (_iterator11.s(); !(_step11 = _iterator11.n()).done;) {
            var _item10 = _step11.value;
            _Certification["default"].validateJsonObject(_item10);
          }
        } catch (err) {
          _iterator11.e(err);
        } finally {
          _iterator11.f();
        }
        ;
      }
      if (data['people_also_viewed']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['people_also_viewed'])) {
          throw new Error("Expected the field `people_also_viewed` to be an array in the JSON data but got " + data['people_also_viewed']);
        }
        // validate the optional field `people_also_viewed` (array)
        var _iterator12 = _createForOfIteratorHelper(data['people_also_viewed']),
          _step12;
        try {
          for (_iterator12.s(); !(_step12 = _iterator12.n()).done;) {
            var _item11 = _step12.value;
            _PeopleAlsoViewed["default"].validateJsonObject(_item11);
          }
        } catch (err) {
          _iterator12.e(err);
        } finally {
          _iterator12.f();
        }
        ;
      }
      // ensure the json data is an array
      if (!Array.isArray(data['recommendations'])) {
        throw new Error("Expected the field `recommendations` to be an array in the JSON data but got " + data['recommendations']);
      }
      if (data['activities']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['activities'])) {
          throw new Error("Expected the field `activities` to be an array in the JSON data but got " + data['activities']);
        }
        // validate the optional field `activities` (array)
        var _iterator13 = _createForOfIteratorHelper(data['activities']),
          _step13;
        try {
          for (_iterator13.s(); !(_step13 = _iterator13.n()).done;) {
            var _item12 = _step13.value;
            _Activity["default"].validateJsonObject(_item12);
          }
        } catch (err) {
          _iterator13.e(err);
        } finally {
          _iterator13.f();
        }
        ;
      }
      if (data['similarly_named_profiles']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['similarly_named_profiles'])) {
          throw new Error("Expected the field `similarly_named_profiles` to be an array in the JSON data but got " + data['similarly_named_profiles']);
        }
        // validate the optional field `similarly_named_profiles` (array)
        var _iterator14 = _createForOfIteratorHelper(data['similarly_named_profiles']),
          _step14;
        try {
          for (_iterator14.s(); !(_step14 = _iterator14.n()).done;) {
            var _item13 = _step14.value;
            _SimilarProfile["default"].validateJsonObject(_item13);
          }
        } catch (err) {
          _iterator14.e(err);
        } finally {
          _iterator14.f();
        }
        ;
      }
      if (data['articles']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['articles'])) {
          throw new Error("Expected the field `articles` to be an array in the JSON data but got " + data['articles']);
        }
        // validate the optional field `articles` (array)
        var _iterator15 = _createForOfIteratorHelper(data['articles']),
          _step15;
        try {
          for (_iterator15.s(); !(_step15 = _iterator15.n()).done;) {
            var _item14 = _step15.value;
            _Article["default"].validateJsonObject(_item14);
          }
        } catch (err) {
          _iterator15.e(err);
        } finally {
          _iterator15.f();
        }
        ;
      }
      if (data['groups']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['groups'])) {
          throw new Error("Expected the field `groups` to be an array in the JSON data but got " + data['groups']);
        }
        // validate the optional field `groups` (array)
        var _iterator16 = _createForOfIteratorHelper(data['groups']),
          _step16;
        try {
          for (_iterator16.s(); !(_step16 = _iterator16.n()).done;) {
            var _item15 = _step16.value;
            _PersonGroup["default"].validateJsonObject(_item15);
          }
        } catch (err) {
          _iterator16.e(err);
        } finally {
          _iterator16.f();
        }
        ;
      }
      // ensure the json data is an array
      if (!Array.isArray(data['skills'])) {
        throw new Error("Expected the field `skills` to be an array in the JSON data but got " + data['skills']);
      }
      // validate the optional field `inferred_salary`
      if (data['inferred_salary']) {
        // data not null
        _PersonEndpointResponseInferredSalary["default"].validateJSON(data['inferred_salary']);
      }
      // ensure the json data is a string
      if (data['gender'] && !(typeof data['gender'] === 'string' || data['gender'] instanceof String)) {
        throw new Error("Expected the field `gender` to be a primitive type in the JSON string but got " + data['gender']);
      }
      // validate the optional field `birth_date`
      if (data['birth_date']) {
        // data not null
        _PersonEndpointResponseBirthDate["default"].validateJSON(data['birth_date']);
      }
      // ensure the json data is a string
      if (data['industry'] && !(typeof data['industry'] === 'string' || data['industry'] instanceof String)) {
        throw new Error("Expected the field `industry` to be a primitive type in the JSON string but got " + data['industry']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['interests'])) {
        throw new Error("Expected the field `interests` to be an array in the JSON data but got " + data['interests']);
      }
      // validate the optional field `extra`
      if (data['extra']) {
        // data not null
        _PersonEndpointResponseExtra["default"].validateJSON(data['extra']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['personal_emails'])) {
        throw new Error("Expected the field `personal_emails` to be an array in the JSON data but got " + data['personal_emails']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['personal_numbers'])) {
        throw new Error("Expected the field `personal_numbers` to be an array in the JSON data but got " + data['personal_numbers']);
      }
      return true;
    }
  }]);
  return PersonEndpointResponse;
}();
/**
 * The vanity identifier of the public LinkedIn profile. The vanity identifier comes after the `/in/` part of the LinkedIn Profile URL in the following format: `https://www.linkedin.com/in/<public_identifier>`
 * @member {String} public_identifier
 */
PersonEndpointResponse.prototype['public_identifier'] = undefined;

/**
 * A temporary link to the user's profile picture that is valid for 30 minutes. The temporal nature of the link is by design to prevent having Proxycurl be the mirror for the images. The developer is expected to handle these images by downloading the image and re-hosting the image. See [this post](https://nubela.co/blog/why-is-the-api-returning-s3-links-for-profile-pictures-scraped-from-linkedin-profiles/) for context.                                  Some profile pictures might be of the standard LinkedIn's profile picture placeholder. It is so because. See [this post](https://nubela.co/blog/why-do-most-linkedin-profiles-fetched-via-the-person-profile-endpoint-return-a-placeholder-profile-picture/) for context.
 * @member {String} profile_pic_url
 */
PersonEndpointResponse.prototype['profile_pic_url'] = undefined;

/**
 * A temporary link to the user's background cover picture that is valid for 30 minutes. The temporal nature of the link is by design to prevent having Proxycurl be the mirror for the images. The developer is expected to handle these images by downloading the image and re-hosting the image. See [this post](https://nubela.co/blog/why-is-the-api-returning-s3-links-for-profile-pictures-scraped-from-linkedin-profiles/) for context.
 * @member {String} background_cover_image_url
 */
PersonEndpointResponse.prototype['background_cover_image_url'] = undefined;

/**
 * First name of the user.
 * @member {String} first_name
 */
PersonEndpointResponse.prototype['first_name'] = undefined;

/**
 * Last name of the user.
 * @member {String} last_name
 */
PersonEndpointResponse.prototype['last_name'] = undefined;

/**
 * Full name of the user (`first_name` + `last_name`)
 * @member {String} full_name
 */
PersonEndpointResponse.prototype['full_name'] = undefined;

/**
 * The title and company name of the user's current employment.
 * @member {String} occupation
 */
PersonEndpointResponse.prototype['occupation'] = undefined;

/**
 * The tagline written by the user for his profile.
 * @member {String} headline
 */
PersonEndpointResponse.prototype['headline'] = undefined;

/**
 * A blurb (longer than the tagline) written by the user for his profile.
 * @member {String} summary
 */
PersonEndpointResponse.prototype['summary'] = undefined;

/**
 * The user's country of residence depicted by a 2-letter country code (ISO 3166-1 alpha-2).
 * @member {String} country
 */
PersonEndpointResponse.prototype['country'] = undefined;

/**
 * The user's country of residence, in English words.
 * @member {String} country_full_name
 */
PersonEndpointResponse.prototype['country_full_name'] = undefined;

/**
 * The city that the user is living at.
 * @member {String} city
 */
PersonEndpointResponse.prototype['city'] = undefined;

/**
 * The state that the user is living at.
 * @member {String} state
 */
PersonEndpointResponse.prototype['state'] = undefined;

/**
 * The user's list of historic work experiences.
 * @member {Array.<module:model/Experience>} experiences
 */
PersonEndpointResponse.prototype['experiences'] = undefined;

/**
 * The user's list of education background.
 * @member {Array.<module:model/Education>} education
 */
PersonEndpointResponse.prototype['education'] = undefined;

/**
 * A list of languages that the user claims to be familiar with, and has added to his/her profile.                                  Do note that we do not have the proficiency level as that data point is not available on a public LinkedIn profile.
 * @member {Array.<String>} languages
 */
PersonEndpointResponse.prototype['languages'] = undefined;

/**
 * List of noteworthy organizations that this user is part of.
 * @member {Array.<module:model/AccomplishmentOrg>} accomplishment_organisations
 */
PersonEndpointResponse.prototype['accomplishment_organisations'] = undefined;

/**
 * List of noteworthy publications that this user has partook in.
 * @member {Array.<module:model/Publication>} accomplishment_publications
 */
PersonEndpointResponse.prototype['accomplishment_publications'] = undefined;

/**
 * List of noteworthy honours and awards that this user has won.
 * @member {Array.<module:model/HonourAward>} accomplishment_honors_awards
 */
PersonEndpointResponse.prototype['accomplishment_honors_awards'] = undefined;

/**
 * List of noteworthy patents won by this user.
 * @member {Array.<module:model/Patent>} accomplishment_patents
 */
PersonEndpointResponse.prototype['accomplishment_patents'] = undefined;

/**
 * List of noteworthy courses partook by this user.
 * @member {Array.<module:model/Course>} accomplishment_courses
 */
PersonEndpointResponse.prototype['accomplishment_courses'] = undefined;

/**
 * List of noteworthy projects undertaken by this user.
 * @member {Array.<module:model/Project>} accomplishment_projects
 */
PersonEndpointResponse.prototype['accomplishment_projects'] = undefined;

/**
 * List of noteworthy test scores accomplished by this user.
 * @member {Array.<module:model/TestScore>} accomplishment_test_scores
 */
PersonEndpointResponse.prototype['accomplishment_test_scores'] = undefined;

/**
 * List of historic volunteer work experiences.
 * @member {Array.<module:model/VolunteeringExperience>} volunteer_work
 */
PersonEndpointResponse.prototype['volunteer_work'] = undefined;

/**
 * List of noteworthy certifications accomplished by this user.
 * @member {Array.<module:model/Certification>} certifications
 */
PersonEndpointResponse.prototype['certifications'] = undefined;

/**
 * Total *count* of LinkedIn connections.
 * @member {Number} connections
 */
PersonEndpointResponse.prototype['connections'] = undefined;

/**
 * A list of other LinkedIn profiles closely related to this user.
 * @member {Array.<module:model/PeopleAlsoViewed>} people_also_viewed
 */
PersonEndpointResponse.prototype['people_also_viewed'] = undefined;

/**
 * List of recommendations made by other users about this profile.
 * @member {Array.<String>} recommendations
 */
PersonEndpointResponse.prototype['recommendations'] = undefined;

/**
 * A list of LinkedIn status activities.
 * @member {Array.<module:model/Activity>} activities
 */
PersonEndpointResponse.prototype['activities'] = undefined;

/**
 * A list of other LinkedIn profiles with similar names.
 * @member {Array.<module:model/SimilarProfile>} similarly_named_profiles
 */
PersonEndpointResponse.prototype['similarly_named_profiles'] = undefined;

/**
 * A list of content-based articles posted by this user.
 * @member {Array.<module:model/Article>} articles
 */
PersonEndpointResponse.prototype['articles'] = undefined;

/**
 * A list of LinkedIn groups that this user is a part of.
 * @member {Array.<module:model/PersonGroup>} groups
 */
PersonEndpointResponse.prototype['groups'] = undefined;

/**
 * A list of keyword-based skills that this user boasts of on his LinkedIn profile.
 * @member {Array.<String>} skills
 */
PersonEndpointResponse.prototype['skills'] = undefined;

/**
 * @member {module:model/PersonEndpointResponseInferredSalary} inferred_salary
 */
PersonEndpointResponse.prototype['inferred_salary'] = undefined;

/**
 * Gender of the user.
 * @member {String} gender
 */
PersonEndpointResponse.prototype['gender'] = undefined;

/**
 * @member {module:model/PersonEndpointResponseBirthDate} birth_date
 */
PersonEndpointResponse.prototype['birth_date'] = undefined;

/**
 * Industry that the user works in.
 * @member {String} industry
 */
PersonEndpointResponse.prototype['industry'] = undefined;

/**
 * A list of interests that the user has.
 * @member {Array.<String>} interests
 */
PersonEndpointResponse.prototype['interests'] = undefined;

/**
 * @member {module:model/PersonEndpointResponseExtra} extra
 */
PersonEndpointResponse.prototype['extra'] = undefined;

/**
 * A list of personal emails associated with this user.
 * @member {Array.<String>} personal_emails
 */
PersonEndpointResponse.prototype['personal_emails'] = undefined;

/**
 * A list of personal mobile phone numbers associated with this user.
 * @member {Array.<String>} personal_numbers
 */
PersonEndpointResponse.prototype['personal_numbers'] = undefined;
var _default = PersonEndpointResponse;
exports["default"] = _default;
