// proxycurl-js-linkedin-profile-scraper/dist/model/Project.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _CompanyUpdatePostedOn = _interopRequireDefault(require("./CompanyUpdatePostedOn"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The Project model module.
 * @module model/Project
 * @version 1.0.0
 */
var Project = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Project</code>.
   * @alias module:model/Project
   */
  function Project() {
    _classCallCheck(this, Project);
    Project.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Project, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Project</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Project} obj Optional instance to populate.
     * @return {module:model/Project} The populated <code>Project</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Project();
        if (data.hasOwnProperty('starts_at')) {
          obj['starts_at'] = _CompanyUpdatePostedOn["default"].constructFromObject(data['starts_at']);
        }
        if (data.hasOwnProperty('ends_at')) {
          obj['ends_at'] = _CompanyUpdatePostedOn["default"].constructFromObject(data['ends_at']);
        }
        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
        if (data.hasOwnProperty('url')) {
          obj['url'] = _ApiClient["default"].convertToType(data['url'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Project</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Project</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `starts_at`
      if (data['starts_at']) {
        // data not null
        _CompanyUpdatePostedOn["default"].validateJSON(data['starts_at']);
      }
      // validate the optional field `ends_at`
      if (data['ends_at']) {
        // data not null
        _CompanyUpdatePostedOn["default"].validateJSON(data['ends_at']);
      }
      // ensure the json data is a string
      if (data['title'] && !(typeof data['title'] === 'string' || data['title'] instanceof String)) {
        throw new Error("Expected the field `title` to be a primitive type in the JSON string but got " + data['title']);
      }
      // ensure the json data is a string
      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      }
      // ensure the json data is a string
      if (data['url'] && !(typeof data['url'] === 'string' || data['url'] instanceof String)) {
        throw new Error("Expected the field `url` to be a primitive type in the JSON string but got " + data['url']);
      }
      return true;
    }
  }]);
  return Project;
}();
/**
 * @member {module:model/CompanyUpdatePostedOn} starts_at
 */
Project.prototype['starts_at'] = undefined;

/**
 * @member {module:model/CompanyUpdatePostedOn} ends_at
 */
Project.prototype['ends_at'] = undefined;

/**
 * Name of the project that has been or is currently being worked on.
 * @member {String} title
 */
Project.prototype['title'] = undefined;

/**
 * Description of the project.
 * @member {String} description
 */
Project.prototype['description'] = undefined;

/**
 * A web location related to the project.
 * @member {String} url
 */
Project.prototype['url'] = undefined;
var _default = Project;
exports["default"] = _default;
