// proxycurl-js-linkedin-profile-scraper/dist/model/PublicationPublishedOn.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ModelDate = _interopRequireDefault(require("./ModelDate"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The PublicationPublishedOn model module.
 * @module model/PublicationPublishedOn
 * @version 1.0.0
 */
var PublicationPublishedOn = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PublicationPublishedOn</code>.
   * Date of publication.
   * @alias module:model/PublicationPublishedOn
   * @implements module:model/ModelDate
   */
  function PublicationPublishedOn() {
    _classCallCheck(this, PublicationPublishedOn);
    _ModelDate["default"].initialize(this);
    PublicationPublishedOn.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(PublicationPublishedOn, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>PublicationPublishedOn</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PublicationPublishedOn} obj Optional instance to populate.
     * @return {module:model/PublicationPublishedOn} The populated <code>PublicationPublishedOn</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PublicationPublishedOn();
        _ModelDate["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('day')) {
          obj['day'] = _ApiClient["default"].convertToType(data['day'], 'Number');
        }
        if (data.hasOwnProperty('month')) {
          obj['month'] = _ApiClient["default"].convertToType(data['month'], 'Number');
        }
        if (data.hasOwnProperty('year')) {
          obj['year'] = _ApiClient["default"].convertToType(data['year'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>PublicationPublishedOn</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>PublicationPublishedOn</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      return true;
    }
  }]);
  return PublicationPublishedOn;
}();
/**
 * @member {Number} day
 */
PublicationPublishedOn.prototype['day'] = undefined;

/**
 * @member {Number} month
 */
PublicationPublishedOn.prototype['month'] = undefined;

/**
 * @member {Number} year
 */
PublicationPublishedOn.prototype['year'] = undefined;

// Implement ModelDate interface:
/**
 * @member {Number} day
 */
_ModelDate["default"].prototype['day'] = undefined;
/**
 * @member {Number} month
 */
_ModelDate["default"].prototype['month'] = undefined;
/**
 * @member {Number} year
 */
_ModelDate["default"].prototype['year'] = undefined;
var _default = PublicationPublishedOn;
exports["default"] = _default;
