// proxycurl-js-linkedin-profile-scraper/dist/model/VolunteeringExperience.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _CompanyUpdatePostedOn = _interopRequireDefault(require("./CompanyUpdatePostedOn"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
/**
 * The VolunteeringExperience model module.
 * @module model/VolunteeringExperience
 * @version 1.0.0
 */
var VolunteeringExperience = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>VolunteeringExperience</code>.
   * @alias module:model/VolunteeringExperience
   */
  function VolunteeringExperience() {
    _classCallCheck(this, VolunteeringExperience);
    VolunteeringExperience.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(VolunteeringExperience, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>VolunteeringExperience</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/VolunteeringExperience} obj Optional instance to populate.
     * @return {module:model/VolunteeringExperience} The populated <code>VolunteeringExperience</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new VolunteeringExperience();
        if (data.hasOwnProperty('starts_at')) {
          obj['starts_at'] = _CompanyUpdatePostedOn["default"].constructFromObject(data['starts_at']);
        }
        if (data.hasOwnProperty('ends_at')) {
          obj['ends_at'] = _CompanyUpdatePostedOn["default"].constructFromObject(data['ends_at']);
        }
        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
        }
        if (data.hasOwnProperty('cause')) {
          obj['cause'] = _ApiClient["default"].convertToType(data['cause'], 'String');
        }
        if (data.hasOwnProperty('company')) {
          obj['company'] = _ApiClient["default"].convertToType(data['company'], 'String');
        }
        if (data.hasOwnProperty('company_linkedin_profile_url')) {
          obj['company_linkedin_profile_url'] = _ApiClient["default"].convertToType(data['company_linkedin_profile_url'], 'String');
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
        if (data.hasOwnProperty('logo_url')) {
          obj['logo_url'] = _ApiClient["default"].convertToType(data['logo_url'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>VolunteeringExperience</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>VolunteeringExperience</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `starts_at`
      if (data['starts_at']) {
        // data not null
        _CompanyUpdatePostedOn["default"].validateJSON(data['starts_at']);
      }
      // validate the optional field `ends_at`
      if (data['ends_at']) {
        // data not null
        _CompanyUpdatePostedOn["default"].validateJSON(data['ends_at']);
      }
      // ensure the json data is a string
      if (data['title'] && !(typeof data['title'] === 'string' || data['title'] instanceof String)) {
        throw new Error("Expected the field `title` to be a primitive type in the JSON string but got " + data['title']);
      }
      // ensure the json data is a string
      if (data['cause'] && !(typeof data['cause'] === 'string' || data['cause'] instanceof String)) {
        throw new Error("Expected the field `cause` to be a primitive type in the JSON string but got " + data['cause']);
      }
      // ensure the json data is a string
      if (data['company'] && !(typeof data['company'] === 'string' || data['company'] instanceof String)) {
        throw new Error("Expected the field `company` to be a primitive type in the JSON string but got " + data['company']);
      }
      // ensure the json data is a string
      if (data['company_linkedin_profile_url'] && !(typeof data['company_linkedin_profile_url'] === 'string' || data['company_linkedin_profile_url'] instanceof String)) {
        throw new Error("Expected the field `company_linkedin_profile_url` to be a primitive type in the JSON string but got " + data['company_linkedin_profile_url']);
      }
      // ensure the json data is a string
      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      }
      // ensure the json data is a string
      if (data['logo_url'] && !(typeof data['logo_url'] === 'string' || data['logo_url'] instanceof String)) {
        throw new Error("Expected the field `logo_url` to be a primitive type in the JSON string but got " + data['logo_url']);
      }
      return true;
    }
  }]);
  return VolunteeringExperience;
}();
/**
 * @member {module:model/CompanyUpdatePostedOn} starts_at
 */
VolunteeringExperience.prototype['starts_at'] = undefined;

/**
 * @member {module:model/CompanyUpdatePostedOn} ends_at
 */
VolunteeringExperience.prototype['ends_at'] = undefined;

/**
 * Name of volunteer activity.
 * @member {String} title
 */
VolunteeringExperience.prototype['title'] = undefined;

/**
 * @member {String} cause
 */
VolunteeringExperience.prototype['cause'] = undefined;

/**
 * The company's display name.
 * @member {String} company
 */
VolunteeringExperience.prototype['company'] = undefined;

/**
 *          The company's profile URL.          If present, could be used with [Company Profile Endpoint](#company-api-company-profile-endpoint) for more info.         
 * @member {String} company_linkedin_profile_url
 */
VolunteeringExperience.prototype['company_linkedin_profile_url'] = undefined;

/**
 * @member {String} description
 */
VolunteeringExperience.prototype['description'] = undefined;

/**
 * URL of the logo of the organisation.
 * @member {String} logo_url
 */
VolunteeringExperience.prototype['logo_url'] = undefined;
var _default = VolunteeringExperience;
exports["default"] = _default;
